---
title: "Cryptic mussels in a contact zone: towards to species distribution modelling"
author: "**V. M. Khaitov, A. A. Zaychikova, P. Y. Safonov, M. V. Katolikova, M. V. Ivanov, P. P. Strelkov**"
output:
  # html_document
  word_document:
    reference_docx: article_template.docx
    fig_width: 5
    fig_height: 5
# bibliography: Astred_english_Bibliografy_abriviated.bib
# csl: marine-biology.csl
---

```{r setup, include=FALSE}
library(knitr)
opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE, dpi = 300)
options(knitr.kable.NA = '')
```

```{r}
## Packages ######

library(ggplot2)
library(lme4)
library(glmmADMB)
library(reshape2)
library(dplyr)
library(patchwork)
library(broom)
library(broom.mixed)
library(tidyr)
library(readxl)
library(mgcv)
library(gratia)
library(flextable)
library(cowplot)

theme_set(theme_bw())

```

```{r training_data}
##### Data reading
myt_full <- read_excel(path = "data/myt_full_2024.xlsx", sheet = "Samples")


ports <- data.frame(Shore = c("Kand", "Karel", "Kand", "Karel", "Karel", "Karel"), Port = c("Kandalaksha", "Vitino", "Umba", "Chupa", "Keret", "Kovda"), Status = c("Active", "Active", "Abandoned", "Abandoned", "Abandoned", "Abandoned"  ), Lat = c(67.137283, 67.076570,  66.677970, 66.269964, 66.294178, 66.696754), Lon = c(32.407995, 32.333630, 34.357655, 33.069534, 33.640656, 32.875396))


# sites_fetch_df <- read.table("data/Distred_samples_fetch_values_2023.csv", sep = ",", header = T)
# 
# sites_fetch_df[,1:5] <- sites_fetch_df[,1:5]/1000 

# Сonventional coordinates of the top of the Kandalaksha Bay
Shore_boundary = c(67.162360, 32.332371)


river_full <- read.table("data/Rivers_2021.csv", sep = ",", header = T)

```



```{r Boundary_of_Ptros_between_MT_and_ME_dominated_sites}

boundary = 0.5

```




```{r}
myt_full <- 
  myt_full %>%
  mutate(Ptros =  exp(-2.4 + 5.4 * Prop_T)/(1 + exp(-2.4 + 5.4 * Prop_T)) ) %>% 
  mutate(Log_Min_dist_port = log(Min_dist_port),
         Log_Min_dist_river = log(Min_dist_river),
         Log_Average_Fetch = log(Average_Fetch),
         Log_Fetch = log(Fetch)) %>% 
  mutate(Mt_dominated = ifelse(Ptros > boundary, 1, 0))
  
```






```{r training_data_housekeepimg}

myt_full$Position <- factor(myt_full$Position)

myt_full$Position <- relevel(myt_full$Position, ref = "Bottom")

myt_full$Port_Status <- factor(myt_full$Port_Status)

myt_full$Port_Status <- relevel(myt_full$Port_Status, ref = "Abandoned")

myt_full$River_Size <- factor(myt_full$River_Size)

myt_full$River_Size <- relevel(myt_full$River_Size, ref = "Small")

myt_full$Site <- factor(myt_full$Site)

# names(myt_full)
```




```{r training_data_sites}
myt_site <- myt_full %>% 
  group_by(Site) %>% 
  # select(Lat, Lon, N_T, N_E, Salinity, Min_dist_river, River, River_Size, Min_dist_river_Large, Min_dist_port, Port, Port_Status, Average_Fetch, Fetch, Dist_cut) %>% 
  summarise(Lat = mean(Lat), Lon = mean(Lon), N_T = sum(N_T), N_E = sum(N_E), Salinity = mean(Salinity), Min_dist_river = mean(Min_dist_river), River = unique(River), River_Size = unique(River_Size), Min_dist_river_Large = mean(Min_dist_river_Large),  Min_dist_port = mean(Min_dist_port), Port = unique(Port), Port_Status = unique(Port_Status), Average_Fetch = mean(Average_Fetch), Fetch = mean(Fetch), Dist_cut = mean(Dist_cut)) %>% 
  mutate(Prop_T = N_T/(N_T+N_E)) %>% 
  mutate(Ptros = exp(-2.4 + 5.4 * Prop_T)/(1 + exp(-2.4 + 5.4 * Prop_T)))

```




```{r training_data_sites_substrates }

myt_site_substr <-
  myt_full %>%
  group_by(Site, Position) %>%
  summarise(N_E = sum(N_E), N_T = sum(N_T),
        Salinity = mean(Salinity),
        Min_dist_river = mean(Min_dist_river),
        River_Size = unique(River_Size),
        Average_Fetch = mean(Average_Fetch),
        Fetch = mean(Fetch),
        Min_dist_port = mean(Min_dist_port),
        Port_Status = unique(Port_Status),
        Lon = mean(Lon),
        Lat = mean(Lat)) %>%
  mutate(Prop_T = N_T/(N_T + N_E)) %>%
  mutate(Ptros =  exp(-2.4 + 5.4 * Prop_T)/(1 + exp(-2.4 + 5.4 * Prop_T))) %>%  
  mutate(Mt_dominated = ifelse(Ptros > boundary, 1, 0))


myt_site_substr <- 
  myt_site_substr %>% 
  mutate(Log_Min_dist_port = log(Min_dist_port),
         Log_Min_dist_river = log(Min_dist_river),
         Log_Fetch = log(Fetch))

```



```{r myt_testing_data_set}
myt_test <- read_excel("data/myt_White_Sea_testing_data_set.xlsx")

myt_test <-
  myt_test %>% 
  filter(Exclude == 0)

myt_test_site <- myt_test %>% group_by(Site, Position) %>%  
  summarise(N_E = sum(N_E), N_T = sum(N_T),
        Salinity = mean(Salinity, na.rm = T),
        Min_dist_river = mean(Min_dist_river),
        River_Size = unique(River_Size),
        Average_Fetch = mean(Average_Fetch),
        Fetch = mean(Fetch),
        Min_dist_port = mean(Min_dist_port),
        Port_Status = unique(Port_Status),
        Lon = mean(Lon),
        Lat = mean(Lat)) %>%
  mutate(Prop_T = N_T/(N_T + N_E)) %>%
  mutate(Fi_T = 2*asin(sqrt(Prop_T))*180/pi) %>%
  mutate(Ptros = exp(-2.4 + 5.4 * Prop_T)/(1 + exp(-2.4 + 5.4 * Prop_T)),
         Fi_tros = 2*asin(sqrt(Ptros))*180/pi) %>% 
  mutate(Mt_dominated = ifelse(Ptros > boundary, 1, 0))

myt_test_site <- 
  myt_test_site %>% 
  mutate(Log_Min_dist_port = log(Min_dist_port),
         Log_Min_dist_river = log(Min_dist_river),
         Log_Average_Fetch = log(Average_Fetch),
         Log_Fetch = log(Fetch))

```





```{r Tyuva_testing_data_set}
tuv <- read_excel("data/TuMyt_2009_2010_for_SDM.xlsx")

tuv <- tuv %>% mutate(Mt_dominated  = ifelse(Ptros_predicted > boundary, 1, 0))

# tuv <- tuv %>% filter(Depth > -1.5)
tuv <-
  tuv %>% mutate(Ptros_WSBL = exp(-2.4 + 5.4 * PT)/(1 + exp(-2.4 + 5.4 * PT)),
                 Ptros = exp(-2.3 + 3.3*PT)  /  (1 +  exp(-2.3 + 3.3*PT)))

tuv <- 
  tuv %>% 
  mutate(Log_Min_dist_port = log(Min_dist_port),
         Log_Min_dist_river = log(Min_dist_river + 1),
         Log_Average_Fetch = log(Average_Fetch),
         Log_Fetch = log(Fetch))
```


```{r}

kola <- read_excel("data/Testing_data_from_Khaitov_et_al_2021.xlsx", sheet = "S1 Kola samples")
kola$Port_Status <- factor(kola$Port_Status)
kola$Position <- "Bottom"
# kola$Position <- "Algae"

kola$Position <- factor(kola$Position)
kola$River_Size <- factor(kola$River_Size)

kola$Log_Fetch <- log(kola$Fetch)
kola$Log_Min_dist_port <- log(kola$Min_dist_port)
kola$Log_Min_dist_river <- log(kola$Min_dist_river)

kola <- 
  kola %>% 
  mutate(Ptros = ifelse(Sample_set == "BL", exp(-2.4 + 5.4 * PT)/(1 + exp(-2.4 + 5.4 * PT)), 
         exp(-3.9 + 5.0 * PT)/(1 + exp(-3.9 + 5.0 * PT)))) %>% 
  mutate(Mt_dominated = ifelse(Ptros > boundary, 1, 0)) %>% 
  filter(Region == "Kola Bay")

```



## Introduction

Species distribution models (SDM) being a numerical tools describing the relationship between species occurrence and environmental parameters  allow to predict distribution patterns of species both in space and time (Elith, Leathwick, 2009). SDMs are usually considered, as well, as a formal way of species' ecological niche assessment  (Elith, Leathwick, 2009; +++) and when the modelling is applied to several coexisting species this approach (Joint Species Distribution Modelling, JSDM) allow to discover the ecological niches partitioning between species (Ovaskainen, Abreg, 2020). The range of approaches to building such models is very wide: from regular multiple regressions up to advanced machine learning methods (+++). The SDM/JSDM framework is very popular in terrestrial ecology (+++) but the amount of papers devoted to SDMs for marine species has also been increasing in recent years (Robinson et al 2017). 

SDMs are mostly applied for «good», morphologically distinct species  (e.g. Reiss et al., 2011; Lindegren et al., 2022) that can be involved in routine studies which in the case of marine ecology  require numerous samples.  The increasing evidence for cryptic species (Bastrop et al 1998; Fiser et al 2010) and polytipic species (++++REF) among marine animals makes it urgent to find out approaches to build SDM (more precisely JSDM) for sympatric cryptic species ~~, as well as to infraspecific taxa such as subspecies or phylogenetic lineages within polytypic species~~.  Only few studies have been conducted in this direction in the marine ecology so far (Lowen et al 2019; Dennis, Hellberg, 2010). 

In the marine realm, the blue mussel "*Mytilus edulis*" complex is the longest scientifically known and best-studied group of cryptic species (Knowlton 1993, ???). This complex includes several species that are more easily distinguished genetically than morphologically and have the ability to hybridize in sympatry (++++++). Blue mussels are powerful ecosystem engineers in the temperate and subpolar seas playing an important role in coastal communities (+++). They are also important objects of aquaculture (+++). In the North Atlantic sector, the dominant species are *M. edulis* (thereafter *ME*) and *M. trossulus* (*MT*), which form zones of sympatry (thereafter *contact zones*) in different parts of the region from Scotland and the Gulf of Maine in the south to Greenland and Svalbard in the north (Wenne et al. 2020 and references therein). In contact zones, *ME*, *MT* and their hybrids are often found in the same samples from mussel settlements; such settlements are hereafter referred to as "*mixed*". Scientists generally agree that *ME* and *MT* are ecologically distinct in sympatry (++++ RC, Katolikova et al.), and have different economic values in aquaculture (+++), but the data on the factors of their ecological segregation i.e. ecological niche partitioning is fragmentary and frequently contradictory.


On a biogeographic scale, the distribution of *Mytilus* species is thought to be regulated primarily by temperature and its correlates (Hayhurst, Rawson, 2009; Wenne et al., 2020). Both species occur in the Arctic, but *MT* being more stenothermic, cold-living species does not penetrate as far south to the temperate/boreal seas as *ME* (Wenne et al., 2020). 

The greatest progress in comparative ecological studies of *ME* and *MT* in sympatry has been made in the contact zones in the Baltic Sea, in the waters of the Kola Peninsula (White and Barents Seas) and in the West Atlantic. In the Baltic Sea, only *MT* inhabits the brackish areas of the inner Sea parts, but the more saline areas closer to the North Sea inhabited by *ME* only. In the contact zone hybrids usually predominate in mixed settlements and the frequency of *MT* genes increases towards the inner Baltic (VS 11, Stuckas et al. 2017). As a result, species distribution is strongly correlated with salinity, against which the role of other factors is negligible (++++). However in the Kola region and in West Atlantic zones the situation differs  from that in the Baltic. Hybridization is limited with less than 20% of hybrids in mixed settlements (Wenne et al. 2020; Katolikova et al. 2016). Species are distributed in space in a mosaic fashion, both in regional and local scales (RC, +++++ ). The relationship between distribution of species and salinity is nowhere obvious (RC 05, Katolikova et al. 2016; Marchenko et al. 2023), but a number of other factors of ecological segregation have been proposed.

In the White and Barents Seas, the frequency of *MT* is elevated in port areas, possibly due to introduction of this species into the region with ship traffic (VS11++++). The only one factor of species segregation explicitly tested in the White Sea was the substrate to which littoral mussels attach (Katolikova et al. 2016). It turned out that *MT* is more common on fucoid algae while *ME* mostly lives directly on the bottom (mud, sand, stones, gravel). The non-random distribution of species across substrates, however, cannot explain the entire local-scale mosaic in their distribution (Katolikova et al. 2016). Parallel studies in the Barents Sea did not consider mussel substrates, but verified the hypotheses of non-random distribution of species with depth. It turned out that on vertical transects, the proportion of *ME* increases with depth and, as a result, *MT* appears to be a more littoral species and *ME* a more sublittoral one (Marchenko et al. 2023). In the western Atlantic, depth, level of anthropogenic pollution and surf exposure has been considered as possible factors in species segregation. It was hypothesized that *MT* may be more abundant than *ME* in littoral (++++), polluted (++++), and surf-exposed habitats (+++), but none of these hypotheses have been tested on sufficient material. 

Summarizing the above, in the non-Baltic *MT*-*ME* contact zones  no clear pattern of species distribution along ecological gradients was revealed and some factors involved into analysis were potentially collinear, masking each other. For example, ports are often located in storm-protected areas, usually in estuaries. This makes it difficult to isolate the effects of shipping and other anthropogenic factors, surf and salinity. The same could be said about the effects of depth and fouling substrate since the littoral fucoids are rare in the sublittoral. This state of knowledge is not surprising given that for most of the time of blue mussel research, scientists have used labor-intensive genotyping methods to identify cryptic species and therefore have been unable to precede large amounts of material. In addition, there were no reliable statistical methods available to model the distribution of sympatric taxa in the space of multiple factors, i.e. no SDM/JSDM approach was realized. To our knowledge, in the history of *ME* and *MT* studies the SDM approach (but without direct reference to this) have been applied only twice, by Kijewski et al.(+++) and by Wenne et al. (+++). Both times the machine learning techniques were used to model the macro-geographic distribution of species (technically, of allele frequencies at taxonomically informative genes) in the space of multiple climatic and oceanographic characteristics available from public databases. The conclusions of these studies are summarized by recognizing temperature and salinity as important factors influencing the geographical distribution with *MT* tolerate lower salinities and temperatures than *ME* (Kijewski et al. +++ Wenne et al. +++, see also above).    


Finding a simple semi-diagnostic shell trait for *ME* and *MT* - the presence or absence of a continuous prismatic strip under the ligament on the inner side of the shell (Zolotarev +++, Katolikova et al. 2016), allowed us to reliably interpret the taxonomic structure of mixed settlements without genotyping. In the White Sea 74% of *MT*, but only 4% of *ME* have a discribed strip (Katolikova et al., 2016), and using the regression models the frequencies of “striped” morphotypes in samples can be recalculated into proportion of *ME* and *MT* (thereafter *ME:MT* ratio; see equations in Khaitov et al., 2021). Note that hybrids are not considered as separate category under this approach.

In this study, we used the SDM approach to simultaneously assess the effects of substrate, salinity, proximity to harbors, and surf on the *ME:MT* ratio in the Kandalaksha Bay of the White Sea. All these factors have been proposed in previous studies as determinants of species' ecological niche partitioning in sympatry. The last of the previously discussed factors, depth, was not studied by us, but was controlled by sampling at the same littoral level. We will also look in more detail at the influence of substrate type as a possible factor in species segregation. Finally, we will attempt to evaluate the ability of the constructed model to predict the *ME:MT* ratio using test material collected from both the White Sea and neighboring Barents Sea contact zones




## Material and methods


```{r material_discription_values}
Total_samples <- nrow(myt_full)
Total_site <- length(unique(myt_full$Site))
Site_published <- length(unique(myt_full$Site [myt_full$Source != "Original"]))

Total_N <- sum(myt_full$N_T) + sum(myt_full$N_E)
Min_N <- min(myt_full$N_T + myt_full$N_E)
Max_N <- max(myt_full$N_T + myt_full$N_E)
Mean_N <- mean(myt_full$N_T + myt_full$N_E)
SD_N <- sd(myt_full$N_T + myt_full$N_E)


```


```{r incomplete_training_sites}
incomplete_sites <- c("Bersakol", "Kovd", "Youzh", "Kuz1", "Chup1", "Chup2", "Chup4", "Chup3", "Por", "Umb", "Gridshel", "Gridexp")

# myt_full <- myt_full %>% filter(! Site %in% incomplete_sites)

```



## Material and methods

### Study area

The 185 km long Kandalaksha Bay is funnel-shaped with numerous islands and skerries and highly indented  coastline (Fig. 1). Climate is continental subarctic with 4-5 months of ice cover and the average monthly sea surface temperature in August of 13.8°C (Filatov et al., 2005). Mean tidal range is about 2 m. Summer surface salinity is 24-25 ppt throughout most of the Bay, but is lower in the estuarine areas of the rivers (++++). About two dozen of rivers and two hydropower plant's canals with a catchment area of 141 - 12830 km^2^ (see Stable ++ ) flow into the Bay, with the largest river, the Niva, entering the Bay at its very top. Due to the complex geometry of the shoreline and numerous rivers, local surf and salinity gradients are pronounced (Fig 1 a, b). 

Mussels are ubiquitous and abundant on the littoral, particularly in the fucoid algal belt, consisting mainly of *Fucus vesiculosus* and *Ascophyllum nodosum* (Berger et al. 2001). 


Historically (through the 20th century), six ports were functioning in the Bay operating oceanic vessels (Fig. 1). Two of these ports, both in the top of the Bay, are still functioning while the remaining are abandoned (Sailing directions of the White Sea, 1932; Krasavtsev, 2011) but still visited by small ships, according to the common knowledge.




### Mussel sampling and processing


Between 2011 and 2018, mussels were sampled at `r Total_site` sites within the littoral fucoid belt (materials from `r Site_published` sites was published previously as S2 Table in Katolikova et al. 2016). Sites were chosen to describe littoral populations in the Bay in as much detail as possible and to account for the heterogeneity of their habitat in terms of substrate type, surf exposure, and distance from rivers and harbors. All samples were taken in the middle part of the fucoid belt to minimize differences in depth. At each site, three samples from fucoid thalluses (hereinafter, *Algal* samples) and three samples from bottom substrates (*Bottom* samples) were collected in few meters from each other using 0.25 m^2^ and 0.025 m^2^ frames respectively. In all cases frame was placed on the bottom area where mussels' aggregations were visually recognized. 


In all samples only mussels with shell size more 10 mm were involved in further analysis. The processing of mussels collected from different substrates was differently organized. From the *Bottom* samples all mussels of appropriated size were selected for further analysis. From *Algal* samples, one bundle of algae (selected so that to contain some dozens of mussels) was taken from the frame and weighted along with attached mussels. The remain algae were taken from the frame and weighed by the same manner (i.e with attached mussels). Only mussels from the bundle selected were used in further analysis. To assess the total number of mussels in the *Algae* sample, the weight of the bundle with the known number of mussels and the weight of the algae rest  in the frame were compared. For `r length(incomplete_sites)` sites the information on abundance of mussels in algal samples was lacking, and these sites were excluded from those analyses which required this data.

Shell  morphotypes (T- and E-morphotype, i.e. mussels with and without of a continuous prismatic strip under the ligament, respecively) were identified as in Khaitov et al. (2021). For further, the frequencies of morphotypes in each sample was converted to the proportion of *MT* (*Ptros*) using the *Eq 1* (Khaitov et al., 2021). The *Eq 1* describes the association of *Ptros*  and proportion of mussel with T-morphotype (*PT*). The value of *Ptros* was calculated for each sample, for each site in total (thereafter *Ptros~Site~*) and for joined samples from  *Algae* and from *Bottom* substrates for each site.

$$
Ptros = \frac{e^{-2.4 + 5.4PT}}{1+e^{-2.4 + 5.4PT}} \ \ \ \ \ Eq \ 1.
$$


### Environmental parameters assessment

In total, we used seven parameters describing possible influence of rivers, ports, surf and type of substrate on mussels (Table ++). The measures of river influence was conducted by three ways. We directly measured salinity in the time of sampling with accuracy of 1 ppm using an “Atago S/Mill-E” refractometer. However, believing that a single estimate of salinity at low tide may not be sufficient to characterize salinity per se, and more generally the extent of estuarine conditions at the sampling sites, we measured  the distance to the nearest river and classified neares rivers by its size on the basis of their catchment areas (ESM STable +++). To calculate Fetch as a proxy of surf exposure, the R-package “windfetch” (Seers, 2022) was used. The high resolution shape-files of regional geographic maps were used for this analysis.  


```{r}
library(officer)
params <- read_excel("data/Model_parameters_discription.xlsx")

params$Range[2] <- paste("Algae VS Bottom")


params$Range[4] <- paste(min(myt_full$Salinity),"-", max(myt_full$Salinity), " (", median(myt_full$Salinity), ")", sep = "") 

params$Range[5] <- paste(round(min(myt_full$Min_dist_river), 1),"-", round(max(myt_full$Min_dist_river), 1), " (", round(median(myt_full$Min_dist_river), 1), ")", sep = "") 

params$Range[6] <- paste("Small VS Large")

params$Range[8] <- paste(round(min(myt_full$Min_dist_port), 1),"-", round(max(myt_full$Min_dist_port), 1), " (", round(median(myt_full$Min_dist_port), 1), ")", sep = "") 

params$Range[9] <- paste("Active VS Abandoned")

params$Range[11] <- paste(round(min(myt_full$Average_Fetch), 1),"-", round(max(myt_full$Average_Fetch), 1), " (", round(median(myt_full$Average_Fetch), 1), ")", sep = "") 


ft <- flextable(params)

column_name <- c(
  "Environmental parameter/ model predictor",
  "Type",
  "Explanation",
  "Range (median) in the data" 
)

ft <-
ft %>% 
set_header_labels(values = column_name) %>% 
  set_caption(caption = "Table ++. Environmental parameters used for the analysis")

ft %>% 
  merge_at(i = 1) %>%
  merge_at(i = 3) %>%
  merge_at(i = 7) %>%
  merge_at(i = 10) %>%
  align(i = c(1, 3, 7, 10), align = "center") %>%
  italic(i = c(1, 3, 7, 10)) %>% 
  border(i = c(1, 3, 7, 10), border.bottom =  fp_border(color = "black")) %>% 
  width(j =  c(1, 2,3,4), width = c(3, 2, 10, 4), unit = "cm")
  
# %>% 
#   autofit()

```



### Testing datasets


Additionally we included into analysis two other data arrays which were used as testing datasets. The first one consisted of samples from `r length(unique(myt_test_site$Site))` sites collected in the Kandalaksha bay (STable ++, Fig. 1 B). These sites were sampled using a different protocol than in the case of main dataset (i.e. only *Bottom* or only *Algae* samples were collected per each site). All predictor's values and *Ptros* for these samples were assessed as for main dataset.

The second testing dataset was  obtained from published data on mussel populations in the Tyuva inlet (Marchenko et al. 2023, supplementary electronic materials, `r length(unique(tuv$Sample_ID))` samples in 2009-2010. This inlet is situated in the Barents Sea where another *MT-ME* contact zone is presented. We used *Ptros* values provided by authors. Predictors were assessed as follow. Since the algae cover was assessed in the sampling area we classified those samples where cover rank was high (4-5 in presented data set) or the samples were taken from kelp forest  as samples from *Algae* substrates. When cover ranks were lower (1-3) the samples were classified as *Bottom*. The Tyuva River flowing into the upper part of the inlet having the catchment area of 351 km^2^ was assessed as a "Large" river (see above). The fetch values for each sites were assessed by the same manner as descrbed above. The nearest port was an active one and was located in the Kola bay  (69.19 N, 33.496 E) about 3 km from the mouth of Tyuva inlet. 

We split the data from the Tyuva inlet into two parts. The first part included sites located 0.5 - 1.5 m above chart datum, i.e. approximatelly in the fucoid belt zone (as was in the main dataset). The second part included sites located above the fucoid belt (2 m upper chart datum) and in the sublittoral (-0.5 to -3.5 m). We analysed these two parts of dataset separately.    



### Statistical analysis


All processing was performed using the statistical programming language R 4.05 (R core Team, 2021). 


Assuming that the relationship between the dependent variable (*Ptros*) and predictors can be curvilinear (Austin, 2002), we used GAM (generalised additive model , Wood, 2017) as a modelling technique that has worked well for SDM construction (Elith et al., 2006). The GAM fitted (thereafter *Model 1*) was based on beta-binomial residuals distribution and the restricted maximum likelihood (REML) method for parameters estimation. Smoothers for all continuous predictors were fitted using cubic basic splines. Categorical predictors were included as parametric terms in the model. "Site" was considered as random factor in the *Model 1*. The function gam() from the package “mgcv” (Wood 2017) was used to fit the model.  

To check for the predictors' collinearaity in the model we calculated the variance inflation factor (VIF, Fox & Monette, 1992). Additionally we calculated Pearson correlation between continuous predictors. To verify that the *Model 1* meet the assumptions of sampling independence, we examined the presence of residuals' spatial autocorrelation by means of spline correlogram construction (Bjornstad, Falck, 2001) with the function spline.correlog() from the package "ncf" (Bjornstad, 2022). No evidences of spatial autocorrelation were revealed for the *Model 1*.

For a more deep analysis of association between *Ptros* and substrate type we calculated the difference (*Dif*) between *Ptros* on *Algae*  and *Bottom* substrates for each site. The obtained *Dif* values were used as dependent variable in the *Model 2* which was constructed as GAM with gaussian residuals' distribution. The proportion of *MT* in site (*Ptros~Site~*) and the mean mussel's abundance in the site per 1 m^2^ (*MeanN*) were used as predictors in *Model 2*. We used function gam() from the package "mgcv" for the model construction. 


To assess ability of *Model 1* to predict the *MT-ME* distribution we used three data sets: (1) main training dataset (the data used for *Model 1* construction), (2) testing dataset from the Kandalaksha bay, (3) the first part of testing dataset from the Tyuva bay, i.e. data collected from fucoid's belt and (4) the second part of data from the Tyuva inlet, i.e. sites out of fucoid's belt. 

To assess the model's predictive power all sites included in the analysis were divided into two groups *MT*-dominated and *ME*-dominated using *Ptros* = 0.5 as a boundary between them. Using the parameters of *Model 1*  we calculated predictions of the model for each of four datasets. Thereafter the predicted values were considered as classifiers for detecting *MT*- or *ME*-dominated groups. The receiver operated characteristics (ROC) followed by analyzing the area under curve (AUC, Fielding, Bell,1997; Fawcett, 2006) was used to evaluate models performance. The function roc() from the package "pROC" (Xavier et al. 2011) was used.





## Results

## Environmental parameters in the Kandalaksha Bay

Two of six ports presented in the area were active at the date of material collection (Fig. 1 A, STable ++ SEM). The most exposed sites (maximum values of wind fetch) were located on the western coast of the Kandalaksha Bay, and on the open shores of the islands at the top of the bay (Fig. 1 A). The most desalinated areas were the top of the Bay and inlets where rivers flow into (Fig. 1 A). Salinity at sites situated close to “Large” rivers was lower than at sites closer to “Small” ones (SFig. ++, C). Salinity tended to increase with increasing of distance to the mouth of the nearest river (SFig. ++ D) but no association between distance to the nearest port and salinity was found (SFig. +++ E). Sites situated close to ports tended to have lower wind fetch (SFig. +++ F). All correlations between salinity, distance to the nearest river mouth, distance to the nearest port and wind fetch were rather low (STable +++). The largest correlation revealed was between wind fetch and distance to nearest port (r = ++).          


![Figure +. Characteristics of sampling sites in the Kandalaksha bay. (A) The surf condition in the sampling sites: point size is proportional to Fetch value. Anchor signs mark position of harbours. Abandoned ports are marked by asterisks. (B) Salinity in the sampling sites. Arrows mark large river's mouths. (C) and (D) *Ptros* in Bottom and Algae samples, respectively. Inset maps are placed to show detailed points in the upper parts of the bay.](figures/Combine_Fetch_Salinity_Ptros_Substrate.png)



## *Ptros* spatial distribution in the Kandalaksha Bay

The maximum *Ptros*  was observed in the top of the Kandalaksha Bay itself and in the small bays deeping into the mainland (Fig. +, C, D). At the same time, sites with a lower *Ptros* were located in the immediate vicinity of sites where high *Ptros* values were represented (Fig. +, C, D), reflecting the high variation of taxonomical structure in small spatial scale. Settlements with the lowest *Ptros* both in *Bottom* and *Algae* samples were represented in the open parts of the North-West coast. 




## Relationship of *Ptros* to environmental parameters


```{r Models_calculations}

Mod_gam <- gam(Ptros ~ s(Salinity, bs = "cs") + s(Log_Min_dist_river, bs = "cs") + s(Log_Fetch, bs = "cs") + s(Log_Min_dist_port, bs = "cs") + Position + River_Size + Port_Status + s(Site, k = Total_site, bs = "re"), method = "REML", family = betar(link = "logit"), data = myt_full )

# summary(Mod_gam)

Mod_gam_ML <- gam(Ptros ~ s(Salinity, bs = "cs") + s(Log_Min_dist_river, bs = "cs") + s(Log_Fetch, bs = "cs") + s(Log_Min_dist_port, bs = "cs") + Position + River_Size + Port_Status + s(Site, k = Total_site, bs = "re"), method = "ML", family = betar(link = "logit"), data = myt_full )
# 
# Mod_gam_ML2 <- gam(Ptros ~ s(Salinity, bs = "cs") + s(Log_Fetch, bs = "cs") + s(Log_Min_dist_port, bs = "cs") + Position + River_Size + Port_Status + s(Site, k = Total_site, bs = "re"), method = "ML", family = betar(link = "logit"), data = myt_full )
# 
# 
# Mod_gam_ML3 <- gam(Ptros ~ s(Salinity, bs = "cs") + s(Log_Min_dist_port, bs = "cs") + Position + River_Size + Port_Status + s(Site, k = Total_site, bs = "re"), method = "ML", family = betar(link = "logit"), data = myt_full )
# 
# Mod_gam_ML4 <- gam(Ptros ~ s(Log_Min_dist_port, bs = "cs") + Position + River_Size + Port_Status + s(Site, k = Total_site, bs = "re"), method = "ML", family = betar(link = "logit"), data = myt_full )
# 
# Mod_gam_ML5 <- gam(Ptros ~ Position + River_Size + Port_Status + s(Site, k = Total_site, bs = "re"), method = "ML", family = betar(link = "logit"), data = myt_full )
# 
# Mod_gam_ML6 <- gam(Ptros ~ Position + Port_Status + s(Site, k = Total_site, bs = "re"), method = "ML", family = betar(link = "logit"), data = myt_full )
# 
# Mod_gam_ML7 <- gam(Ptros ~ Position + s(Site, k = Total_site, bs = "re"), method = "ML", family = betar(link = "logit"), data = myt_full )
# 
# Mod_gam_ML8 <- gam(Ptros ~ s(Site, k = Total_site, bs = "re"), method = "ML", family = betar(link = "logit"), data = myt_full )


# summary(Mod_gam_ML7)

# AIC(Mod_gam_ML, Mod_gam_ML2, Mod_gam_ML3, Mod_gam_ML4, Mod_gam_ML5, Mod_gam_ML6, Mod_gam_ML7, Mod_gam_ML8)
# AIC(Mod_gam_ML, Mod_gam_reduced_ML)

# appraise(Mod_gam)

# appraise(Mod_gam_reduced)

# summary(Mod_gam)

# summary(Mod_gam_reduced)

# draw(Mod_gam)

```

```{r Model_prediction}
logit_back <- function(x) exp(x)/(1 + exp(x)) 

# Smoothers estimations

sm <- smooth_estimates(Mod_gam) %>%
  add_confint()

sm <- sm %>% 
  mutate(Pi = logit_back(est + coef(Mod_gam)[1]), 
         CI_Pi_low = logit_back(lower_ci + coef(Mod_gam)[1]),
         CI_Pi_up = logit_back(upper_ci + coef(Mod_gam)[1]))

myt_full <- myt_full %>% 
  add_partial_residuals(Mod_gam)

# Recalculation of smoothers into probability
myt_full <- 
  myt_full %>% 
  mutate(Pi_Salinity = logit_back(`s(Salinity)` + coef(Mod_gam)[1]),
         Pi_Min_dist_river = logit_back(`s(Log_Min_dist_river)` + coef(Mod_gam)[1]),
         Pi_Average_Fetch = logit_back(`s(Log_Fetch)` + coef(Mod_gam)[1]),
         Pi_Min_dist_port = logit_back(`s(Log_Min_dist_port)` + coef(Mod_gam)[1]))



```


```{r Smoother_predictions_figures}
#Predicted values in the terms of smoothers

Pl_sal_smoother <-
  sm %>%
  filter(smooth == "s(Salinity)") %>%
  ggplot() +
  geom_point(aes(x = Salinity, y = `s(Salinity)`, color = Position),
             data = myt_full, cex = 1, colour = "steelblue3") +
  geom_rug(aes(x = Salinity),
           data = myt_full,
           sides = "b", length = grid::unit(0.02, "npc")) +
  geom_ribbon(aes(ymin = lower_ci, ymax = upper_ci, x = Salinity),
              alpha = 0.2) +
  geom_line(aes(x = Salinity, y = est), lwd = 1) +
  labs(y = "Partial effect", x = "Salinity") +
  # geom_hline(yintercept = 0, linetype = 2) +
  theme_bw()


Pl_riv_dist_smoother <- 
    sm %>%
  filter(smooth == "s(Log_Min_dist_river)") %>%
  ggplot() +
  geom_point(aes(x = Log_Min_dist_river, y = `s(Log_Min_dist_river)`, color = Position),
             data = myt_full, cex = 1, colour = "steelblue3") +
  geom_rug(aes(x = (Log_Min_dist_river)),
           data = myt_full,
           sides = "b", length = grid::unit(0.02, "npc")) +
  geom_ribbon(aes(ymin = lower_ci, ymax = upper_ci, x = Log_Min_dist_river),
              alpha = 0.2) +
    geom_line(aes(x = Log_Min_dist_river, y = est), lwd = 1) +
  labs(y = "Partial effect", x = "Log Distance to the nearest river (km)") +
  # geom_hline(yintercept = 0, linetype = 2) +
  theme_bw()
# +
#   ylim(-0.5, 0.5)



  

Pl_port_dist_smoother <-
     sm %>%
  filter(smooth == "s(Log_Min_dist_port)") %>%
  ggplot() +
  geom_point(aes(x = Log_Min_dist_port, y = `s(Log_Min_dist_port)`, color = Position),
             data = myt_full, cex = 1, colour = "steelblue3") +
  geom_rug(aes(x = Log_Min_dist_port),
           data = myt_full,
           sides = "b", length = grid::unit(0.02, "npc")) +
  geom_ribbon(aes(ymin = lower_ci, ymax = upper_ci, x = Log_Min_dist_port),
              alpha = 0.2) +
    geom_line(aes(x = Log_Min_dist_port, y = est), lwd = 1) +
  labs(y = "Partial effect", x = "Log Distance to the nearest port (km)") +
  # geom_hline(yintercept = 0, linetype = 2) +
  theme_bw() 
    

Pl_fetch_smoother <-
  sm %>%
  filter(smooth == "s(Log_Fetch)") %>%
  ggplot() +
  geom_point(aes(x = Log_Fetch, y = `s(Log_Fetch)`, color = Position),
             data = myt_full, cex = 1, colour = "steelblue3") +
  geom_rug(aes(x = Log_Fetch),
           data = myt_full,
           sides = "b", length = grid::unit(0.02, "npc")) +
  geom_ribbon(aes(ymin = lower_ci, ymax = upper_ci, x = Log_Fetch),
              alpha = 0.2) +
    geom_line(aes(x = Log_Fetch, y = est), lwd = 1) +
  labs(y = "Partial effect", x = "Log Fetch (km)") +
  # geom_hline(yintercept = 0, linetype = 2) +
  theme_bw()
  


##################################
#Predicted values in the terms of probabilities

# Pl_sal_smoother_Pi <-
#   sm %>%
#   filter(smooth == "s(Salinity)") %>%
#   ggplot() +
#   geom_point(aes(x = Salinity, y = Pi_Salinity), data = myt_full, cex = 1, colour = "steelblue3") +
#   geom_rug(aes(x = Salinity),data = myt_full, sides = "b", length = grid::unit(0.02, "npc")) +
#   geom_ribbon(aes(ymin = CI_Pi_low, ymax = CI_Pi_up, x = Salinity),alpha = 0.2) +
#   geom_line(aes(x = Salinity, y = Pi), lwd = 1) +
#   labs(y = "Partial effect", x = "Salinity (PPT)") +
#   # geom_hline(yintercept = median(myt_full$Pi_Salinity), linetype = 2)+
#   theme_bw()+
#   ylim(0, 0.8)
# 
# 
# Pl_riv_dist_smoother_Pi <-
#   sm %>%
#   filter(smooth == "s(Log_Min_dist_river)") %>%
#   ggplot() +
#   geom_point(aes(x = Log_Min_dist_river, y = Pi_Min_dist_river), data = myt_full, cex = 1, colour = "steelblue3") +
#   geom_rug(aes(x = Log_Min_dist_river),data = myt_full, sides = "b", length = grid::unit(0.02, "npc")) +
#   geom_ribbon(aes(ymin = CI_Pi_low, ymax = CI_Pi_up, x = Log_Min_dist_river),alpha = 0.2) +
#   geom_line(aes(x = Log_Min_dist_river, y = Pi), lwd = 1) +
#   labs(y = "Partial effect", x = "Distance to the nearest river (km)") +
#   # geom_hline(yintercept = median(myt_full$Pi_Min_dist_river), linetype = 2) +
#   theme_bw()+
#   ylim(0, 0.8)
# 
# 
# 
# Pl_port_dist_smoother_Pi <-
#   sm %>%
#   filter(smooth == "s(Min_dist_port)") %>%
#   ggplot() +
#   geom_point(aes(x = Min_dist_port, y = Pi_Min_dist_port), data = myt_full, cex = 1, colour = "steelblue3") +
#   geom_rug(aes(x = Min_dist_port),data = myt_full, sides = "b", length = grid::unit(0.02, "npc")) +
#   geom_ribbon(aes(ymin = CI_Pi_low, ymax = CI_Pi_up, x = Min_dist_port),alpha = 0.2) +
#   geom_line(aes(x = Min_dist_port, y = Pi), lwd = 1) +
#   labs(y = "Partial effect", x = "Log Distance to the nearest port (km)") +
#   # geom_hline(yintercept = median(myt_full$Pi_Min_dist_port), linetype = 2)+
#   theme_bw()+
#   ylim(0, 0.8)
# 
# 
# Pl_fetch_smoother_Pi <-
#   sm %>%
#   filter(smooth == "s(Fetch)") %>%
#   ggplot() +
#   geom_point(aes(x = Fetch, y = Pi_Average_Fetch), data = myt_full, cex = 1, colour = "steelblue3") +
#   geom_rug(aes(x = Fetch),data = myt_full, sides = "b", length = grid::unit(0.02, "npc")) +
#   geom_ribbon(aes(ymin = CI_Pi_low, ymax = CI_Pi_up, x = Fetch),alpha = 0.2) +
#   geom_line(aes(x = Fetch, y = Pi), lwd = 1) +
#   labs(y = "Partial effect", x = "Log Fetch (km)") +
#   # geom_hline(yintercept = median(myt_full$Pi_Average_Fetch), linetype = 2)+
#   theme_bw()+
#   ylim(0, 0.8)

```




<!-- ```{r} -->
<!-- draw(Mod_gam, parametric = T) -->
<!-- ``` -->


```{r Parametric_prediction_figures}

# Predictions for parametric terms 

ds <-  data_slice(Mod_gam, Position = evenly(Position),  Port_Status = evenly(Port_Status), River_Size = evenly(River_Size)) 


Fit <- predict.gam(Mod_gam, newdata = ds, se = T, type = "link", exclude = c("s(Salinity)", "s(Log_Min_dist_river)",  "s(Log_Average_Fetch)",  "s(Log_Min_dist_port)", "s(Site)"))

ds$Fit = Fit$fit

ds$SE = Fit$se.fit
ds$CI_low <- (Fit$fit - 1.96*Fit$se.fit)
ds$CI_up <- (Fit$fit + 1.96*Fit$se.fit)




ds$Pi <- logit_back(Fit$fit)
ds$Pi_CI_low <- logit_back(Fit$fit - 1.96*Fit$se.fit)
ds$Pi_CI_up <- logit_back(Fit$fit + 1.96*Fit$se.fit)
#
#
Pl_Small <-
ds %>% filter(River_Size == "Small") %>%
ggplot(., aes(x = Position, y = Fit, fill = Port_Status)) +
  geom_col(position = position_dodge()) +
  scale_fill_manual(values = c("blue", "gray")) +
  geom_errorbar(aes(ymin = CI_low, ymax = CI_up), width = 0.2, position = position_dodge(width = 0.9)) +
  theme_bw()+
  ggtitle("The nearest River: Small") +
  theme(legend.position = c(0.8, 0.2)) +
  labs(x = "", y = "Partial effect", fill = "Nearest port status")
#
Pl_Large <-
ds %>% filter(River_Size == "Large") %>%
ggplot(., aes(x = Position, y = Fit, fill = Port_Status)) +
  geom_col(position = position_dodge()) +
  scale_fill_manual(values = c("blue", "gray")) +
  geom_errorbar(aes(ymin = CI_low, ymax = CI_up), width = 0.2, position = position_dodge(width = 0.9)) +
  theme_bw()+
  ggtitle("The nearest River: Large") +
  guides(fill = "none") +
  theme(legend.position = c(0.19, 0.5)) +
  labs(x = "", y = "Partial effect")
#





Pl_Small_Pi <-
ds %>% filter(River_Size == "Small") %>%
ggplot(., aes(x = Position, y = Pi, fill = Port_Status)) +
  geom_col(position = position_dodge()) +
  scale_fill_manual(values = c("blue", "gray")) +
  geom_errorbar(aes(ymin = Pi_CI_low, ymax = Pi_CI_up), width = 0.2, position = position_dodge(width = 0.9)) +
  theme_bw()+
  ggtitle("The nearest River: Small") +
  ylim(0, 0.8)+
  theme(legend.position = c(0.25, 0.8)) +
  labs(x = "", y = "Partial effect", fill = "Nearest port status")
#
Pl_Large_Pi <-
ds %>% filter(River_Size == "Large") %>%
ggplot(., aes(x = Position, y = Pi, fill = Port_Status)) +
  geom_col(position = position_dodge()) +
  scale_fill_manual(values = c("blue", "gray")) +
  geom_errorbar(aes(ymin = Pi_CI_low, ymax = Pi_CI_up), width = 0.2, position = position_dodge(width = 0.9)) +
  theme_bw()+
  ggtitle("The nearest River: Large") +
  ylim(0, 0.8)+
  guides(fill = "none") +
  theme(legend.position = c(0.19, 0.8)) +
  labs(x = "", y = "Partial effect")
    

# appraise(Mod_gam)

# tidy(Mod_gam, parametric = T)

```


```{r Models_summary}
sum_Model_1 <- summary(Mod_gam)

# sum_Model_2 <- summary(Mod_gam_reduced)

Mod_dummy <- lm(Prop_T ~ Salinity + Min_dist_river + Fetch + Min_dist_port + Position + River_Size + Port_Status,  data = myt_full )

library(car)

VIF <- vif(Mod_dummy) 

```



The values of VIF calculated for the predictors' set included in the analysis were rather small (the maximal VIF was detected for `r paste(names(VIF)[which.max(VIF)])`: `r round(max(VIF), 2)`). Although some pairwise correlation  between environmental factors were found (see above) the low VIF values can be interpreted as a negligible collinearity between predictors (i.e. they don't mask influence of each for other).   

To assess the dependency of *Ptros* on all predictors considered we constructed the *Model 1* (Table 1, R^2^ = `r round(sum_Model_1$r.sq, 2)`) which explained  `r round(sum_Model_1$ dev.expl*100, 1)`% of total deviance. The *Model 1* revealed statistically significant dependency of *Ptros* on all predictors except one.

The effective degrees of freedom (*edf*, Table ++) for most smoothers were equal to 1, reflecting the linear dependence of *Ptros* on such continuous predictors as distance to the nearest port (*DistPort*) and fetch (*Fetch*). The exception was *Salinity*, where a pronounced curvilinear dependence was revealed.     





<!-- ```{r} -->
<!-- library(caret) -->

<!-- # gam(Ptros ~ s(Salinity, bs = "cs") + s(Log_Min_dist_river, bs = "cs") + s(Log_Fetch, bs = "cs") + s(Log_Min_dist_port, bs = "cs") + Position + River_Size + Port_Status + s(Site, k = Total_site, bs = "re"), method = "REML", family = betar(link = "logit"), data = myt_full ) -->

<!-- gam_caret <- train(factor(Mt_dominated) ~ Salinity + Log_Min_dist_river + Log_Fetch + Log_Min_dist_port + Position + River_Size + Port_Status, method = "glm", data = myt_site_substr) -->

<!-- # summary(gam_caret) -->

<!-- varImp(gam_caret, scale = F) -->

<!-- ``` -->




```{r Model1_output}
# Model 1 summary output preparation

Model_1_summ_smoothers <- tidy(Mod_gam)

Model_1_summ_smoothers$term <- c(
  "s(Salinity)", 
  "s(DistRiver)",
  "s(Fetch)",
  "s(DistPort)",
  "Random effect s(Site)")
Model_1_summ_smoothers[, 2:4] <- round(Model_1_summ_smoothers[, 2:4], 1)
Model_1_summ_smoothers[, 5] <- round(Model_1_summ_smoothers[, 5], 4)


Model_1_summ_parametric <- tidy(Mod_gam, parametric = T)

Model_1_summ_parametric$term <- c("(Intercept)",
                                  "Substrate(Algae)",
                                  "RiverSize(Large)",
                                  "PortStatus(Active)")
Model_1_summ_parametric[, 2:4] <- round(Model_1_summ_parametric[, 2:4], 1)
Model_1_summ_parametric[, 5] <- round(Model_1_summ_parametric[, 5], 4)

names(Model_1_summ_smoothers) <- names(Model_1_summ_parametric)


all_Model_1_summ <- rbind(Model_1_summ_smoothers, Model_1_summ_parametric)




ft_Model_1_summ_smoothers <- flextable(Model_1_summ_smoothers)

header <- c("Smoother terms", "edf", "ref.edf", "Chi.sq", "p-value")

ft_Model_1_summ_smoothers <- 
  ft_Model_1_summ_smoothers %>% 
  set_header_labels(values = header)






ft_Model_1_summ_parametric <- flextable(Model_1_summ_parametric)
header2 <- c("Parametric terms", "Parameter estimate", "SE", "z-statistic", "p-value")

ft_Model_1_summ_parametric <- 
  ft_Model_1_summ_parametric %>% 
  set_header_labels(values = header2)




```



```{r Model1_print_smoothers}
ft_Model_1_summ_smoothers <-
  ft_Model_1_summ_smoothers %>% 
  set_caption(caption = "Table 1 . Parameters of smoothers and coefficients of parametric terms for the Model describing dependency of Ptros on predictors.")

ft_Model_1_summ_smoothers %>% 
  width(j = 1:5, width = c(8, 4,2,2,2), unit = "cm")
```



```{r Model1_print_parametric}
# ft_Model_1_summ_parametric <-
  # ft_Model_1_summ_parametric %>% 
  # set_caption(caption = "Table + B. Parametric terms for the Model describing dependency of PT on predictors.")

ft_Model_1_summ_parametric %>%   
width(j = 1:5, width = c(8, 4,2,2,2), unit = "cm")
```





```{r Model1_prediction_Figure, fig.height= 8,fig.width=8, fig.cap= "Fig. 3. Predictions of the *Model 1* describing the dependency of *Ptros* on distance to the nearest port (A), average fetch (B), distance to the nearest river (C), salinity (E) and substrate type, nearest port status and size of the largest river (E, F). The  gray ribbons and wiskers demonstrate 95% confidence interval. The dotted lines in panels A-D represent the median predicted value, this value is given to track the trend in the change in model predictions. Points on panels A-D reflect partial residuals but not raw data for each sample."}

plot_grid(Pl_port_dist_smoother, Pl_fetch_smoother, Pl_riv_dist_smoother,  Pl_sal_smoother, Pl_Small_Pi, Pl_Large_Pi,  ncol = 2, labels = "AUTO")

```



<!-- Ниже приведен тот же самый рисунок, но по оси OY отложены значения смузеров не преобразованные в вероятности. Обычно именно именно так рисуют такие визуализации. Здесь понятно как трактовать значение смузера равное нулю (зона предиктора, когда виляние отсутствует. Если бы значимой связи не было бы то линия регрессии легла бы параллельно OX и шли бы в районе нуля). Если перобразовать значение смузра в веротяность, то не очень ясно, что взять в качестве аналога нуля. Я привел в качестве такой точки отсчета медиану предсказанных значений. Вообще-то это надо как-то объяснять. Поэтому, если ты готов понять суть такого рода картики, то предлагаю оставить вариант, котороый пиведен ниже.   -->


<!-- ```{r Model1_prediction_Figure, fig.height= 8,fig.width=8, fig.cap= "Fig. 3. Predictions of the *Model 1* describing the dependency of *Ptros* on distance to the nearest port (A), fetch (B), distance to the nearest river (C), salinity (E) and substrate type, nearest port status and size of the largest river (E, F). The  gray ribbons and wiskers demonstrate 95% confidence interval. The dotted horizontal lines in panels A-D represent the zero smoother's value. Points on panels A-D reflect partial residuals but not raw data for each sample."} -->

<!-- plot_grid(Pl_port_dist_smoother, Pl_fetch_smoother, Pl_riv_dist_smoother,  Pl_sal_smoother, Pl_Small_Pi, Pl_Large_Pi,  ncol = 2, labels = "AUTO") -->

<!-- ``` -->



According to the *Model 1* (Table 1, Fig. 3)  *Ptros* has a tendency to decrease with increasing of distance to the nearest port (Fig. 1 A). The nearest port status had also significant influence on *Ptros*: the value was higher when port status was *Active* (Fig. 3 E, F). The dependency of *Ptros* on fetch (Fig. 3, B) was negative. No significant association between *Ptros* and the distance to the neares  river was was found (Table 2, Fig 3 C).    


```{r Salinity_discription_values }
# Different values needed for salinity effect discription

sal_smoother <- 
sm %>% 
  filter(smooth == "s(Salinity)")  

sm_low_salinity <- 
# sal_smoother %>% filter(est > median(est)) %>% filter(Salinity < 20) 
sal_smoother %>% filter(est > 0) %>% filter(Salinity < 20) 

  
Min_salinity <- round(max(sm_low_salinity$Salinity))

sm_high_salinity <- 
sal_smoother %>% filter(est > 0) %>% filter(Salinity > 20) 

Max_salinity <- round(min(sm_high_salinity$Salinity))


Number_sites_low_salinity <- myt_site %>% filter(Salinity <= round(Min_salinity, 0)) %>% nrow()

Number_sites_high_salinity <- myt_site %>% filter(Salinity >= round(Max_salinity, 0)) %>% nrow()

```



The curvilinear dependence of *Ptros* on salinity (Fig. 3, D) can be described as follows. The predicted value of *Ptros* decrease when salinity moving from minimal values to apr 20-22 ppt when it reached minimum. When salinity increases after this point the *Ptros* value increase as well. 

<!-- However, at low (less than `r round(Min_salinity,0)` ppt) and high (more than `r round(Max_salinity, 0)` ppt) salinity, *Ptros* becomes higher. -->


<!-- This low and high salinity (when partial effect for the *Model 1* higher than zero)  was observed at `r Number_sites_low_salinity` and `r Number_sites_high_salinity` sites respectively (Fig. 1 B). For other `r Total_site - Number_sites_low_salinity - Number_sites_high_salinity` sites, with intermediate salinity, partial effect was less than zero (Fig. 3 D). All low salinity sites were represented in the top parts of different inlets (Fig. 1 B) but no clear pattern revealed in the distribution of sites with high salinity (Fig. 1 B).    -->

According to the *Model 1*  the value of *Ptros* is dramatically higher on the *Algae* substrate than on the *Bottom* one (Table 1, Fig. 3 E, F). This value is higher in those sites which is close to *Active* ports than to *Abandoned* ones. Finally, the *Ptros* value is slightly higher at sites situated close to Large rivers than to Small ones. 



## Dependency of *Ptros* on substrate type


```{r Data_for_Substrate_modelling}

myt_sample_area <- read_excel("Data/myt_full.xls", sheet = "Sampling area assesment", na = "NA")

myt_sample_area <- myt_sample_area %>% dplyr::select(Site, Sample, Position, SQM_factor)


myt_full_reduced <- myt_full %>% filter(! Site %in% incomplete_sites)

myt_full_reduced <-
merge(myt_full_reduced, myt_sample_area)



myt_site_substr_reguced <- 
  myt_full_reduced %>% 
  mutate(D_T = N_T * SQM_factor, D_E = N_E*SQM_factor) %>% 
  group_by(Site, Position) %>% 
  summarise(D = mean(D_E + D_T), N_T = sum(N_T), N_E = sum(N_E),
        Salinity = mean(Salinity),
        Min_dist_river = mean(Min_dist_river),
        River_Size = unique(River_Size),
        Average_Fetch = mean(Average_Fetch),
        Min_dist_port = mean(Min_dist_port),
        Port_Status = unique(Port_Status),
        Lon = mean(Lon),
        Lat = mean(Lat), 
        D_T = mean(D_T),
        D_E = mean(D_E)) %>% 
  mutate(Prop_T = N_T/(N_T + N_E)) %>%
  mutate(Ptros = exp(-2.4 + 5.4 * Prop_T)/(1 + exp(-2.4 + 5.4 * Prop_T))) %>%
  mutate(Fi_T = 2*asin(sqrt(Prop_T))*180/pi)  




df1 <-
myt_site_substr_reguced %>% 
  dplyr::select(Site, Position, Ptros) %>%
  dcast(., formula = Site ~ Position, value.var = c("Ptros") ) 


names(df1) <- c("Site", "Ptros_Bottom", "Ptros_Algae" )
  

df2 <-
myt_site_substr_reguced %>% 
  dplyr::select(Site, Position, D) %>%
  dcast(., formula = Site ~ Position, value.var = c("D") )


names(df2) <- c("Site", "D_Bottom", "D_Algae" )

df3 <-
myt_site_substr_reguced %>% 
  dplyr::select(Site, Position, N_T, N_E) %>% 
  group_by(Site) %>% 
  summarise(N_T = sum(N_T), N_E = sum(N_E)) %>% 
  mutate(Total_Prop_T = N_T/(N_T + N_E)) %>% 
  mutate(Total_Ptros = exp(-2.4 + 5.4 * Total_Prop_T)/(1 + exp(-2.4 + 5.4 * Total_Prop_T))) %>%
  dplyr::select(-N_T, -N_E)


df4 <-
myt_site_substr_reguced %>% 
  group_by(Site) %>% 
  summarise(D_E = mean(D_E, na.rm = TRUE), D_T = mean(D_T, na.rm = TRUE))



all_sites <- merge(df1, df2) %>% merge(., df3) %>% merge(., df4) %>% 
    mutate(Dif_Ptros = Ptros_Algae - Ptros_Bottom) %>% 
  mutate(D_mean = (D_Bottom + D_Algae)/2) %>% 
  mutate(Dif_D = log(D_Algae/D_Bottom)) %>% 
  mutate(D_total = D_T + D_E)

all_sites <- 
  all_sites %>%
  mutate(Log_D_total = log(D_total))

# qplot(y = all_sites$Dif_Ptros, x = all_sites$Total_Ptros) + geom_smooth()


```


```{r Model_Substrate_PT_Difference}

Mod_Dif <- gam(Dif_Ptros ~ s(Total_Ptros, bs = "cs") + s(Log_D_total, bs = "cs"), method = "REML", data = all_sites)

# appraise(Mod_Dif)

# draw(Mod_Dif, residuals = TRUE)

summ_Mod_Dif <- tidy(Mod_Dif)


sm <- smooth_estimates(Mod_Dif) %>%
  add_confint() 

all_sites <-  add_residuals(data = all_sites,model = Mod_Dif, value = "Residuals")


Pl_Dif_model <- 
sm %>% 
  filter(smooth == "s(Total_Ptros)") %>% 
  ggplot(., aes(x = Total_Ptros, y = est)) +
  geom_line(color = "blue", linewidth = 1) +
  geom_ribbon(aes(ymin = lower_ci, ymax = upper_ci), alpha=0.2 ) +
  # geom_hline(yintercept = median(sm$est), linetype = 2) +
  geom_point(data = all_sites, aes(y = Residuals),cex = 1, colour = "steelblue3") +
  labs(x = "Ptros \n in Site", y = "Partial effect")
 

Pl_Dif_model_N <- 
sm %>% 
  filter(smooth == "s(Log_D_total)") %>% 
  ggplot(., aes(x = Log_D_total, y = est)) +
  geom_line(color = "blue", linewidth = 1) +
  geom_ribbon(aes(ymin = lower_ci, ymax = upper_ci), alpha=0.2 ) +
  # geom_hline(yintercept = median(sm$est), linetype = 2) +
  geom_point(data = all_sites, aes(y = Residuals),cex = 1, colour = "steelblue3") +
  labs(x = "Log Mussel abundance \n in Site", y = "Partial effect")

# draw(Mod_Dif, residuals = TRUE)

```


```{r Plot_Model_Dif, fig.cap="Fig. 4. The dependence of difference between *PT* on Algae and Bottom subatrates on Ptros~Site~ (A) and total mussel abundance (B)."}

plot_grid(Pl_Dif_model, Pl_Dif_model_N, labels = "AUTO")

```

The difference between *Ptros* on *Algae* and *Bottom* substrates was significantly dependent on *Ptros~Site~*, i.e. prevalence of *MT* (STable +, Fig. 4 A). The dependency revealed was curvelinear (edf = `r round(summ_Mod_Dif$edf[1],1)`, STable +) with minimal *Dif* values in sites with minimal and maximal values of *Ptros~Site~*. The maximal *Dif* values were associated with intermediate *Ptros~Site~*. This pattern reflect the prevalence of *MT* at *Algae* substrate and prevalence of *ME* at *Bottom* substrate in those sites where  both species are presented in equal amount. No significant dependency was revealed with total absolute mussel abundance (STable +, Fig. 4 B).






```{r ROC_analysis}
library(pROC)

# Mod 1 prediction as classifier 
Mod_gam_prediction_train <- predict(Mod_gam, newdata = myt_site_substr, type = "response", exclude = "s(Site)")

roc_Mod_gam_train <- roc(response = myt_site_substr$Mt_dominated, predictor = Mod_gam_prediction_train)
# plot(roc_Mod_gam_train)



Mod_gam_prediction_test <- predict(Mod_gam, newdata = myt_test_site, type = "response", exclude = "s(Site)")

roc_Mod_gam_test <- roc(response = myt_test_site$Mt_dominated, predictor = Mod_gam_prediction_test)
# plot(roc_Mod_gam_test)




# tuv2 <- 
#   tuv %>%
#   filter(Depth > -1.5 )


tuv2 <-
  tuv %>%
  filter(Depth >= 0.5 & Depth <= 1.5  )
  
tuv3 <-
  tuv %>%
  filter(Depth < 0.5 |  Depth > 1.5 )


Mod_gam_prediction_tuv <- predict(Mod_gam, newdata = tuv, type = "response", exclude = "s(Site)")

roc_Mod_gam_tuv <- roc(response = tuv$Mt_dominated, predictor = Mod_gam_prediction_tuv)



Mod_gam_prediction_tuv2 <- predict(Mod_gam, newdata = tuv2, type = "response", exclude = "s(Site)")

roc_Mod_gam_tuv2 <- roc(response = tuv2$Mt_dominated, predictor = Mod_gam_prediction_tuv2)



Mod_gam_prediction_tuv3 <- predict(Mod_gam, newdata = tuv3, type = "response", exclude = "s(Site)")

roc_Mod_gam_tuv3 <- roc(response = tuv3$Mt_dominated, predictor = Mod_gam_prediction_tuv3)




# Mod_gam_prediction_kola_bottom <- predict(Mod_gam, newdata = kola, type = "response", exclude = "s(Site)")
# 
# kola$Position <- "Algae"
# kola$Position <- factor(kola$Position)

Mod_gam_prediction_kola <- predict(Mod_gam, newdata = kola, type = "response", exclude = "s(Site)")

# 
# Mod_gam_prediction_kola <- (Mod_gam_prediction_kola_algae + Mod_gam_prediction_kola_bottom)/2


roc_Mod_gam_kola <- roc(response = kola$Mt_dominated, predictor = Mod_gam_prediction_kola)



```










## Assessment of model predictive power



```{r Confusion_matrix_calculation}

# Landis & Koch (1977) have suggested the following ranges of agreement for
# the Kappa statistic: poor K < 0.4; good 0.4 < K < 0.75 and excellent K < 0.75 (Fielding & Bell 1997).



# For Training data set
a <- sum(myt_site_substr$Ptros > boundary & Mod_gam_prediction_train > boundary)

b <- sum(myt_site_substr$Ptros < boundary & Mod_gam_prediction_train > boundary)

c <- sum(myt_site_substr$Ptros > boundary & Mod_gam_prediction_train < boundary)

d <- sum(myt_site_substr$Ptros < boundary & Mod_gam_prediction_train < boundary)

N = nrow(myt_site_substr)

Kappa_train = ((a + d) - (((a + c)*(a + b) + (b + d)*(c + d))/N))/(N - (((a+c)*(a+b) + (b+d)*(c+d))/N))

Overall_diagnostic_power_train <- (b + d)/N

Correct_classification_rate_train <- (a + d)/N

Sensitivity_train <- a/(a + c)

Specificity_train <- d/(b + d)

PPP_train <-  a/(a + b)
NPP_train <- d/(c + d)
Misclassification_rate_train <- (b + c)/N



# For Testing data set
a <- sum(myt_test_site$Ptros > boundary & Mod_gam_prediction_test > boundary)

b <- sum(myt_test_site$Ptros < boundary & Mod_gam_prediction_test > boundary)

c <- sum(myt_test_site$Ptros > boundary & Mod_gam_prediction_test < boundary)

d <- sum(myt_test_site$Ptros < boundary & Mod_gam_prediction_test < boundary)

N = nrow(myt_test_site)

Kappa_test = ((a + d) - (((a + c)*(a + b) + (b + d)*(c + d))/N))/(N - (((a+c)*(a+b) + (b+d)*(c+d))/N))

Overall_diagnostic_power_test <- (b + d)/N

Correct_classification_rate_test <- (a + d)/N

Sensitivity_test <- a/(a + c)

Specificity_test <- d/(b + d)

PPP_test <-  a/(a + b)
NPP_test <- d/(c + d)
Misclassification_rate_test <- (b + c)/N



# For Tyuva data set
a <- sum(tuv$Ptros > boundary & Mod_gam_prediction_tuv > boundary)

b <- sum(tuv$Ptros < boundary & Mod_gam_prediction_tuv > boundary)

c <- sum(tuv$Ptros > boundary & Mod_gam_prediction_tuv < boundary)

d <- sum(tuv$Ptros < boundary & Mod_gam_prediction_tuv < boundary)

N = nrow(tuv)

Kappa_tuv = ((a + d) - (((a + c)*(a + b) + (b + d)*(c + d))/N))/(N - (((a+c)*(a+b) + (b+d)*(c+d))/N))

Overall_diagnostic_power_tuv <- (b + d)/N

Correct_classification_rate_tuv <- (a + d)/N

Sensitivity_tuv <- a/(a + c)

Specificity_tuv <- d/(b + d)

PPP_tuv <-  a/(a + b)
NPP_tuv <- d/(c + d)
Misclassification_rate_tuv <- (b + c)/N




# For Tyuva data set Fucoids
a <- sum(tuv2$Ptros > boundary & Mod_gam_prediction_tuv2 > boundary)

b <- sum(tuv2$Ptros < boundary & Mod_gam_prediction_tuv2 > boundary)

c <- sum(tuv2$Ptros > boundary & Mod_gam_prediction_tuv2 < boundary)

d <- sum(tuv2$Ptros < boundary & Mod_gam_prediction_tuv2 < boundary)

N = nrow(tuv)

Kappa_tuv2 = ((a + d) - (((a + c)*(a + b) + (b + d)*(c + d))/N))/(N - (((a+c)*(a+b) + (b+d)*(c+d))/N))

Overall_diagnostic_power_tuv_2 <- (b + d)/N

Correct_classification_rate_tuv_2 <- (a + d)/N

Sensitivity_tuv_2 <- a/(a + c)

Specificity_tuv_2 <- d/(b + d)

PPP_tuv_2 <-  a/(a + b)
NPP_tuv_2 <- d/(c + d)
Misclassification_rate_tuv_2 <- (b + c)/N



# For Tyuva data set out Fucoid belt
a <- sum(tuv3$Ptros > boundary & Mod_gam_prediction_tuv3 > boundary)

b <- sum(tuv3$Ptros < boundary & Mod_gam_prediction_tuv3 > boundary)

c <- sum(tuv3$Ptros > boundary & Mod_gam_prediction_tuv3 < boundary)

d <- sum(tuv3$Ptros < boundary & Mod_gam_prediction_tuv3 < boundary)

N = nrow(tuv)

Kappa_tuv_3 = ((a + d) - (((a + c)*(a + b) + (b + d)*(c + d))/N))/(N - (((a+c)*(a+b) + (b+d)*(c+d))/N))

Overall_diagnostic_power_tuv_3 <- (b + d)/N

Correct_classification_rate_tuv_3 <- (a + d)/N

Sensitivity_tuv_3 <- a/(a + c)

Specificity_tuv_3 <- d/(b + d)

PPP_tuv_3 <-  a/(a + b)
NPP_tuv_3 <- d/(c + d)
Misclassification_rate_tuv_3 <- (b + c)/N



```


```{r Predicted_vs_observed_PT_plots, fig.cap="Fig. 5. Comparison of *Ptros*-values predicted by *Model 1* and obsreved *Ptros* for training data used for the model fitting (A), testing data from the White sea (B) and testing data from the Tyuva inlet colleced in and out of the fucoid's belt (C and D respectively). The horizontal and vertical solid lines reflects the boundary between *MT*- and *ME*-dominated sites. Dotted lines represent the position of perfect fit when observed and predicted values are equal. Labels in the plot corners mark the quadrants with false positive (FP), true positive (TP), true negative (TN) and false negative (FN) predictions."}


Pl_training <-
  qplot(x = myt_site_substr$Ptros,  y = Mod_gam_prediction_train) +
  labs(x = "Obseved Ptros", y = "Predicted Ptros") +
  geom_point(shape = 21, fill = "gray")  +
  scale_fill_manual(values = c("black", "white")) +
  guides(fill= "none") +
  geom_abline(linetype = 2) +
  geom_vline(xintercept = boundary) +
  geom_hline(yintercept = boundary) +
  xlim(0,1) +
  ylim(0,1) +
  geom_label(aes(x = c(0, 0, 1, 1), y = c(0, 1, 1, 0), label = c("TN", "FP", "TP", "FN") ), alpha = 0.5 ) +
  ggtitle("Training data")
  
  
Pl_Kand_testing <-
  qplot(x = myt_test_site$Ptros,  y = Mod_gam_prediction_test) +
  labs(x = "Obseved Ptros", y = "Predicted Ptros") +
  geom_point(shape = 21, fill = "gray") +
  scale_fill_manual(values = c("white", "black")) +
  guides(fill= "none") +
  geom_abline(linetype = 2) +
  geom_vline(xintercept = boundary) +
  geom_hline(yintercept = boundary) +
  xlim(0,1) +
  ylim(0,1) +
  geom_label(aes(x = c(0, 0, 1, 1), y = c(0, 1, 1, 0), label = c("TN", "FP", "TP", "FN") ), alpha = 0.5 ) +
  ggtitle("Kandalksha bay testing data")

markers <- data.frame(x = c(0, 0, 1, 1), y = c(0, 1, 1, 0), label = c("TN", "FP", "TP", "FN"))
  

tuv2$Mod_gam_prediction_tuv <- Mod_gam_prediction_tuv2

Pl_Tuva_testing_2 <-
  ggplot(tuv2, aes(x = Ptros_predicted,  y = Mod_gam_prediction_tuv2)) +
  labs(x = "Obseved Ptros", y = "Predicted Ptros") +
  geom_point(shape = 21, fill = "gray") +
  scale_fill_manual(values = c("white", "black")) +
  guides(fill= "none") +
  geom_abline(linetype = 2) +
  geom_vline(xintercept = boundary) +
  geom_hline(yintercept = boundary) +
  xlim(0,1) +
  ylim(0,1) +
  geom_label(data = markers, aes(x, y, label = label), alpha = 0.5 ) +
  ggtitle("Tyuva inlet testing data Part 1") 



tuv3$Mod_gam_prediction_tuv <- Mod_gam_prediction_tuv3

Pl_Tuva_testing_3 <-
  ggplot(tuv3, aes(x = Ptros_predicted,  y = Mod_gam_prediction_tuv3)) +
  labs(x = "Obseved Ptros", y = "Predicted Ptros") +
  geom_point(shape = 21, fill = "gray") +
  scale_fill_manual(values = c("white", "black")) +
  guides(fill= "none") +
  geom_abline(linetype = 2) +
  geom_vline(xintercept = boundary) +
  geom_hline(yintercept = boundary) +
  xlim(0,1) +
  ylim(0,1) +
  geom_label(data = markers, aes(x, y, label = label), alpha = 0.5 ) +
  ggtitle("Tyuva inlet testing data Part 2") 



Pl_Kola_testing <-
  qplot(x = kola$Ptros,  y = Mod_gam_prediction_kola) +
  labs(x = "Obseved Ptros", y = "Predicted Ptros") +
  geom_point(shape = 21, aes(fill = kola$Position)) +
  scale_fill_manual(values = c("white", "black")) +
  guides(fill= "none") +
  geom_abline(linetype = 2) +
  geom_vline(xintercept = boundary) +
  geom_hline(yintercept = boundary) +
  xlim(0,1) +
  ylim(0,1) +
  geom_label(aes(x = c(0, 0, 1, 1), y = c(0, 1, 1, 0), label = c("TN", "FP", "TP", "FN") ), alpha = 0.5 ) +
  ggtitle("Tyuva inlet testing data")


  
plot_grid(Pl_training, Pl_Kand_testing, Pl_Tuva_testing_2, Pl_Tuva_testing_3, labels = "AUTO")
# 
# library(patchwork)
# 
# (Pl_training + Pl_Kand_testing)/Pl_Tuva_testing



```



The *Ptros* values  predicted by the *Model 1*  and  observed ones (Fig. 5) were in rather good agreement  for training dataset (AUC = `r round(as.numeric(roc_Mod_gam_train$auc),2)`). A part of points were presented in the false negative zone (the *Model 1* predicts presence *ME*-dominated populations, but *MT* - dominated ones were observed). A few points are situated in the false negative zone. 

In the case of testing dataset from the Kandalaksha bay (Fig. 1 B) the most sites were correctly classified by the *Model 1* (AUC = `r round(as.numeric(roc_Mod_gam_test$auc),2)`). Two points were located in false negative and only one in false positive zone. 


The *Model 1* being applied to the first part of testing dataset from the Tyuva inlet (i.e. collections from fucoid's belt) display lesser but still rather high predictive power (AUC = `r round(as.numeric(roc_Mod_gam_tuv2$auc),2)`). Five samples were classified as false positive, but only one as false negative. The predictive power of the *Model 1*  being applied to the second part of the Tyuva testing dataset was similar to the previous case (AUC = `r round(as.numeric(roc_Mod_gam_tuv3$auc),2)`). Four samples were classified as false positive and only one as false negative. 





## Discussion




Elith et al. 2006
AUC values can be interpreted as indicating the probability that, when a presence site and
an absence site are drawn at random from the population, the first will have a higher predicted value than the second.


better than random (AUC/0.5)

The grey vertical solid line shows random predictions, and grey dashed indicates
reasonable predictive performance.

<!-- Все факторы, будучи включенными в одну модель действительно влияют. -->
<!-- Соленость - универсальный фактор -->
<!-- Фукус - грунт. Предсказываем, что они везде должны расходиться. -->



Processing a large number of samples from various sites across a wide range of environmental conditions in the Kandalaksha Bay contact zone, we developed *Model 1*. This model can be considered both as SDM describing the association with environmental predictors and JSDM reflecting the joint distribution of the two mussel species, *Mytilus edulis* and *M. trossulus*. It is important to acknowledge that the model we have constructed represents only one of the initial efforts to characterize the relationship between cryptic mussel species and their environment. Similar to previous studies by Kijewsky et al.(+++)  and Winne et al. (+++), our analysis focused on the distribution of mussels in their contact zones exclusively. The broader pattern of species distributions, including their relationships with various factors across the entire range of the cryptic species, falls outside the scope of this study. Nevertheless, the most significant events related to species interactions with environmental factors and with each other (including intraspecific relationships) occur primarily in contact zones.


The set of environmental predictors included in the *Model 1*  obtained most  of those factors that were generally considered as the important regulators of the taxonomic structure of mussel's populations in *MT-ME* contact zones (+++++). Being included in one SDM these factors possessed significant influence on *Ptros*  (i.e. proportion of *MT* ). Thus, our analysis provided the consistent evidence that factors such as salinity, proximity to harbours, level of wave action and substrate type influence the taxonomic composition of mixed mussel populations. In addition, we found that the predictors included in the model are not collinear. This means that we can consider all these factors as relatively independent ecological axis that do not mask each other, which is important for SDM construction (De Marco and Nobrega, 2011). In addition we considered environment variation in a relatively small contact zone in the Kandalaksha Bay. We can state that the patterns revealed on more extensive contact zones (e.g. in the Baltic - North Sea contact zone, see Kijewsky et al, 2019) with longer ecological gradients retain their influence in essentially smaller spatial scales. This fact allow to optimistically expect the ability to construct working SDM describing distribution of *MT* and *ME* in contact zones of different sizes.


The SDMs are frequently considered as models reflecting structure of species' ecological niche (Elith, Leathwick, 2009). Although the very concept of 'niche' is under the  criticism (McInerny, Etienne, 2012 a,b,c) analysing the relationship of cryptic species to environmental factors and investigation of mechanismes of their distribution along environmental gradients is undoubtedly an important task. In this sense, any statistically significant association of *Ptros* with some environmental predictor would reflect the divergence of *MT* and *ME* in ecological space (i.e. partitioning of niches). 

From all environmental variables included into analysis the distance to the nearest port and its status (*Abandoned* VS *Active*) may have dual interpretation. We revealed higher *MT* proportion close to active ports. On the one hand, active ports tend to have a higher degree of anthropogenic pollution, leading to stress of mussels (Hellou & Law, 2003). Different mussel species are known to interact differently with pollutants (Beyer et al., 2017). For example, in the Halifax area *MT* are more abundant in the most polluted habitats than *ME*, but show less stress tolerance than *ME* (Hellou & Law, 2003). ~~However, studies of the immune systems of *MT* and *ME* in the Canadian contact zone suggest a greater suppression of the *MT* immune system in areas prone to pollution (Coray et al., 2006), making them more sensitive to stress than *ME*. The last fact contradicts the hypothesis of better performance of *MT* to pollutant influence allowing them to set populations in active port harbors.~~ 
############
Another explanation may be that there is some foci of disease associated with port areas that affect *MT* and *ME* diferently. For example we can look for the reason in such infections as the bivalve transmittable neoplasia which is associated with harbors (Hammel et al., 2023) or in other parasite diseases for which ports are hot-spots for invasion (Feis et al., 2019).

From the other hand the dependency of *PT* on distance to the nearest port and its status can reflect not true axis in ecological space but some trace of historical processes. It is believed that *MT* was introduced into the Barents and the White Seas due to ship traffic during World War II (Vainola & Strelkov, 2011). This hypothesis was based primarily on the fact that high concentrations of *MT* were found in the vicinity of currently active or abandoned ports in the region (Vainola & Strelkov, 2011; Katolikova et al. 2016). The results of our study are inline with this hypothesis. The *Ptros* value  increased with approaching to harbors in the studied area. Apparently, the degree of port activity also plays an important role. According to our data, if the port nearest to the mixed settlement is active, i.e. it serves the ships coming from outside the White Sea (in our case it is "Kandlaksha" and "Vitino" ports), then the *Ptros* value in such settlements is higher than in the settlements adjacent to the abandoned harbors.  The observed relationship may be related to the fact that the currently active ports were the largest and most economically sustainable systems in the region. This may be based on the higher shipping traffic, which has led to a more massive invasion of *MT* from other regions (primarily from the ports of Murmansk in the Kola Bay, where the abundance of *MT* is high, Vainola&Strelrov, 2013; Marchenko et al., 2023). In the abandoned ports, the introduction of the alien species occurred during the times of their active operation. However, during several decades of inactivity of abandoned ports *MT* were probably displaced by native *ME*.

########################


<!-- In any case, our data are in line with earlier stated hypothesis that in the case of the White Sea  *MT* is an invasive species (Vainola&Strelkov 2011). Unfortunately, only few data could be provided to check this hypothesis for other European *MT* -*ME* populations. However the ship traffic was successfully included into SDMs describing invasive species in European waters (Lindegren et al., 2022) this factor was never included into models describing *MT*-*ME* distribution previously. In some cases it is probably an unrealistic aim due to absence of this predictor variability.  For example in the case of *MT*-*ME* populations in the Baltic Sea, where there is extremely active ship traffic and vessels visiting almost all parts of the Baltic Sea, it is not possible to identify a correlation between the distribution of *MT* and the presence of ports. At the same time, if we look at the more broad map of *MT* distribution (Vainola & Strelkov, 2011) around European coasts, the confinement of *MT* to large European ports out of the Baltic Sea (e.g. Bergen, Murmansk), rather speaks in favor of the hypothesis of *MT* introduction due to ship traffic.  Thus, while proximity to ports is not an environmental factor in a direct sense, it is a parameter that should be kept in mind when building a more general SDM describing *MT*-*ME* distribution. -->




<!-- According to the data we obtained, the highest *MT* concentration should be expected on the fuchcoid tally, in sheltered areas close to active ports. All these foundings reflect the pattern of the ecological niche segregation between two sympatricaly coexisting mussel' species.  -->


<!-- Our analysis revealed that the higher wind fetch is associated with lower *PT* value. That means *MT* avoid open coasts potentially exposed to the surf. The lowest *PT* value (i.e., E-morphotype mussels dominate in the settlements) is observed on open shores of the north-east coast of Kandalaksha Bay (Fig. 1 A), exposed to wave action due to southeast and south-west winds which are frequent in the White Sea region (Berger, Naumov, 2001). This result corresponds well with the data on the thickness of *MT* and *ME* shells. It is known that *MT* have thinner, more fragile shells (Beaumont et al., 2008; Michalek et al, 2021; our unpublished observations of mussels from the White Sea are in line with this data). It is logical to expect that mollusks with such characteristics should avoid places with high wave loading. In such habitats, *ME* with stronger shells are expected to benefit. A similar pattern can be seen in other contact zones. For example, in Greenland, an increased frequency of *MT* was noted at the sheltered top of the fjord (GLS and GLD sites, Fig. 1 in Wenne et al. 2016), while the population at a more open site located near fjord mouth (GLL site in Wenne et al. 2016) is dominated by *ME*. -->

<!-- Again, experimental evidence is needed to confirm the divergence of the niches of the two species along the surf-impact axis. It is far from certain that a thinner but more flexible shell could be a hindrance to set settlement in areas more exposed to surf. There is a need to assess the level of mortality of the two mussel species due to shell damage along the surf gradient.      -->

<!-- The nature of the substrate was discovered as the powerful predictor regulating *PT* variation in the scale of several meters. The proportion of T-morphotype mussels appears to be significantly higher on algal than on bottom substrates (silty-sandy sediment, pebbles, boulders and rock surfaces) located in close proximity (not more than 20 m, within the same fucoid belt). This pattern was first detected in the analysis of a significantly smaller volume of material (Katolikova et al., 2016). Bringing in new data collected in a wider range of conditions confirmed the presence of this pattern.  -->

<!-- However, a closer examination of the relationship with substrate type (Fig. 4 A) showed that the maximum *PT* divergence between Algae and Bottom substrates occurred only when both mussel species were present in approximately equal numbers at a given location. This could be interpreted as divergence of realized niches as a result of interspecific competition. This, however, is contradicted by the lack of correlation of *Dif*-value with the absolute abundance of mussels at the site. If the mentioned divergence of realized niches were a consequence of competition, one would expect that the strength of this interaction would increase with increasing settlement density, which is not observed.  Thus another explanation should be considered.  -->

<!-- As possible explanation we can propose some difference in byssus attachment strength. Perhaps *MT* are capable of attaching to underwater substrates more strongly than *ME* do. This should cause less strongly attached molluscs to break off and remain on the bottom under constant fluctuations of fucoid's thallus. This hypothesis is contradicted by the fact that a comparison of attachment strength in *MT* and *ME* from the contact zone in Canada, showed no significant differences between the species when they were kept under laboratory  conditions (Lowen et al., 2013). Thus some additional investigations of attachment strength of both species are needed.  -->


<!-- An intriguing result of our study was the absence of clear linear dependency between taxonomic structure of mixed populations and  salinity that was revealed in other contact zones (Kijewsky, 2019). It is generally considered that *MT* is more tolerant to reduced salinity, where it dominates over *ME*. As salinity increases, species change places (Riginos Cunningham 2005). However, surprisingly, in spite of the fact that salinity is considered as a leading factor determining the segregation of *MT* and *ME* in their contact zones majority of studies discussing this relationship are correlative. Only a few works have been devoted to comparison of ecophysiological responses of these two species to salinity (Gardner&Thompson 2001; Qiu et al., 2002; Knobel et al. 2021). Moreover, the direct physiological investigations did not reveal unequivocal evidence that adult *MT*  are better adapted to reduced salinity than adult *ME* (Gardner&Thompson 2001; Qiu et al., 2002). It has only been shown that larval stages of *MT* survive lower salinity better than the same *ME* stages (Qiu et al., 2002). That is the prevalence of *MT* in lower salinity could be explained by early selective pressure during passive larval drift across salinity gradient (Knobel et al. 2021). -->

<!-- In our investigation the GAM fitted reveal a statistically significant curvelinear (close to U-shape) relationship. From one hand, our *Model 1* revealed higher *PT* (i.e. proportion of *MT*) in lower salinity which is typical for other contact zones. However, on the other hand, this proportion was remarkable inflated when salinity was unusually high. We can propose three explanation to this fact.  -->

<!-- Firstly, although directly measured salinity seems to be a natural ecological factor, this may not be the case for littoral mussel settlements.  The salinity levels significantly vary during the tidal cycle and the range of the variation is highly dependent on the distance from the fresh water source (Attrill, 2002; Marchenko et al., 2023). We do not know which salinity is more important for littoral mussels living in the estuaries: high during the ebb time or reduced in the low tide. Further, the salinity measured once during the sampling process  may occasionally be anomalously low or  high by accident (e.g. due to precipitation or upwelling of deep salted water, respectively). If this explanation is correct, then the observed increase in *PT* at high salinity may be a consequence of some incorrect estimates of salinity values.   -->

<!-- A second explanation for the *PT* peak in high salinity sites may be related to the lower reliability of the morphotype-test for species identification under high salinity conditions. Khaitov et al (2021) suggested that in cold Arctic waters with high salinity, *ME* more often exhibit T-morphotype, which reduces the sensitivity of the method. However this is contradicted by the fact that in the same locations where anomalously high salinity and high *PT* values were detected (e.g. in the Umba area, Fig. 1 A, B) genetic analysis of mussel populations revealed an increased frequency of *MT* genes (Katolikova et al.2016).    -->



<!-- <!-- the number of extreme low salinity sites where *PT* was found to be high is relatively small. Similarly, the number of sites where salinity is very high and where *PT* is also high is very small as well. In total, the number of such anomalous sites is about `r round( (Number_sites_low_salinity + Number_sites_high_salinity)/Total_site *100, 1)` % of the total number of sites surveyed.   --> -->

<!-- And finally, it should be noted that sites with high *PT* and extreme low (<`r Min_salinity` ppt) or unusual high (>`r Max_salinity` ppt) salinity are not numerous. In total, the number of such anomalous sites is about `r round( (Number_sites_low_salinity + Number_sites_high_salinity)/Total_site *100, 1)` % of the total number of sites surveyed. In locations, where salinity is intermediate, *PT* does not differ significantly from the background value. That is, anomalously high *PT* values are observed in some marginal habitats. Such pattern can be expected if we assume that *MT* is a weaker competitor, which on average cannot displace  *ME*. This assumption is supported by the fact that the settlements that we classify as *MT*-dominated are essentially scarcer  in the studied area (TP + FN part of the Fig. 5 A) than the  *ME*-dominated sites (FP + TN part of the Fig. 5 A).  The roughly U-shape association between *PT* and salinity can be turned into bell-like dependency of proportion of E-morphotype (i.e. *ME* frequency) on salinity. The *PT* proportion predicted by the *Model 1* for the  most frequent salinity values (`r Min_salinity` -  `r Max_salinity` ppt) is about 0.1. This means that the majority of mussels presented in the sites within this salinity range are mussels with E-morphotype (mostly *ME*). Such displacement of weaker competitor into marginal habitat is well known for other systems where inviders met resident species:  mosquito larvae (Juliano, 1998), fishes(Herbold, Moyle, 1986) and geckos (Case et al 1994). Usually resident species persist in their optimal habitats if they have not been destroyed but introduced species prevail in marginal habitats (Juliano, 1998). To note, actively used harbors appear to be the very marginal habitats as well.  -->



<!-- All these remarks about salinity make us doubt the reliability of salinity as a good predictor in SDM for littoral mussels. Probably the assessment of salinity influence can be improved when we use more accurate assessment of salinity variation in the site of samples. Undoubtedly salinity is the most important factor regulating the distribution of *MT* and *ME* and this factor be it properly assessed should be taken into account. However the salinity in the form used in our work may not be as high for SDM construction. After we removed salinity from the model, the *Model 2* constructed had AIC very close to AIC of the *Model 1*. The rule of thumb in the model selection paradigm (Burnham, Anderson 2004) suggests that an AIC delta less than 2 indicates high support for the reduced model, that is, the information loss when salinity was dropped from the model is not critical. The comparison of full *Model 1* and reduced *Model 2* by their predictive capacity by ROC analysis did not reveal significant decreasing  of predictive power of *Model 2* when it was applied to training dataset.  -->

<!-- The strongest argument that the constructed model is useful is its ability to predict phenomena on the dataset that was not used to build the model. We used two testing datasets. One was represented by material collected in the same area as the training samples (Fig. 5 B). The analysis of ROC-curves showed that *Model 2* has a high ability to predict the assignment of the sample to one of the two categories *MT*-dominated vs *ME*-dominated. When this model was used for predictions in a different region, the Tyuva inlet, where an independent contact zone is presented, the quality of the prediction decreased. However, two points should be emphasized. Firstly, the AUC=  for *Model* 2 was quite high and close to a reasonable predictive performance AUC level (Elith et al. 2006). Secondly, it is important to note that the predictive power of the *Model 1*  which includes salinity was lower, further emphasizing the low power of this parameter as a predictor of *MT* and *ME* distribution. -->

<!-- The important result of the *Model* 2 applying was the high number of false positive predictions in the case of the Tyuva inlet. The prediction errors may have numerous reasons (see Fielding & Bell, 1997). Firstly it can be consequence of a poor model tuning. In fact, applying the *Model 2* we made some assumptions which may not be entirely realistic. For example we considered samples with high algae coverage as samples from fucoid's tallus. Additionally  the small area of the inlet did not provide sufficient variation in some parameters which is based on distances (fetch measure, distance to the nearest port or river mouth). Although there may be some ecological reasons as well. If a suitable habitat is not colonized by a target species which population is small (i.e. the area is not saturated) or the population is collapsed due to displacement as a result of inter-specific competition, all this reasons could produce false positive result when a model-classifier is applied (Fielding & Bell, 1997). Indeed the *MT* population in the Tyuva inlet displayed  pronounced decreasing associated with increased *ME* population (Marchenko et al 2023). We may expect the Tyuva inlet conditions are highly suitable for *MT* population  but once appeared it demonstrated decrease probably due to competitive displacement by *ME*. We expect that alternative scenario may take place in the case of the Kandalaksha Bay. Let's remember that *MT*-dominated sites were quite rare in the area and that most of false positive prediction were made for Algae substrates (which is more suitable for *MT*). It can be interpreted as evidence of underdevelopment of *MT* population in the area: suitable habitats (primarily fucoids) are not occupied still. It can be hypothesized that this is the result of recent *MT* population formation in the Kandalaksha Bay and some outbreak of *MT* invasion may happen in the area. -->

<!-- Whether our model can predict the distribution of *MT* and *ME* in other contact zones is an unsolved  propblem still and requires additional data. At the same time, it can be noted that in the case of the famous  sudden appearance of *MT* in a region previously dominated by *ME* - the Loch Etive farms (Beaumont et al. 2008), our model seems to be quite applicable at least at the conceptual level. Indeed, the aquaculture farms where unexpected invasion of *MT* was described (56° 27′ 30″ N, 5° 19′ 12″ W,  Beaumont et al. 2008) are located in a narrow fjord (low fetch) with high salinity variability associated with the influx of the full-flowing  Etive river (salinity ranges from 1 to 22 ppt, Beumont et al, 2008), and an active port in Oban located about 14 km along the coastline. All this correspondences indicate that this area could be predicted by our model as the "red zone" where *MT* may establish their populations with high probability. -->


<!-- ## Conclusion -->

<!-- In our analysis, we did not aim to construct some perfect SDM it is the task for further investigation when information on different contact zones would be combined. That is why we did not try to build some final model (for example, by backward selection procedure) and did not try to find the most appropriate modelling method (see Elith et al 2006). It was important for us to understand whether in principle a working model can be built based on those predictors that have come to the attention of researchers studying contact zones of cryptic mussel species.  Our results are optimistic about this possibility but since the contact zones between *MT* and *ME* are scattered over a very wide range of latitudes and situated in various biotic surroundings the appropriate form of the "general" model should be well considered. It is obvious that some additional predictors (e. g. temperature or its correlates, the position in the intertidal-subtidal gradient, *etc*) should be included in a future model. It is also clear that the "general" model should  include some biotic interactions (see Wisz et al 2013) and be developed within the JSDM framework (see Ovaskainen & Abrego, 2021 for a theory). The dependent part of this model should include not only proportions of two coexisting mussel species but a matrix describing the community where mussel's species coexist. Our study has shown that algae play an important role in the local distribution of species. Perhaps the spatial variation in algae abundance (and probably algae species composition) can play a significant role in the distribution of the cryptic mussel species. Thus algae characteristics in region should be somehow incorporated in the model. Perhaps some role may also be played by predators, which can regulate the proportion of mussel species (Khaitov et al.2018; Khaitov et al.2023 ). However the first step to construct the working SDM is the deep comparison of ecological features of different *MT-ME* contact zones.   -->




 
## Literature cited 

Лоция Белого моря / Гидрографическое управление ; общ. ред. А. Н. Рождественского. – 5-е изд. – Ленинград : Издание Гидрографического управления, 1932. – LXXII, 566 с., 22 вкл. л. ил., карт., схем.




\pagebreak

# Supplementary electronic materials (SEM)

<!-- ### Environmental parameters characteristics  -->

<!-- <!-- There are 6 ports presented in Kandalaksha Bay (STable +++). Most localities sampled (34 sites) had "Vitino" as the closest harbor, which came into service in the 1990s. The remaining ports, in operation since the beginning of the 20th century, were attributed as closest to 14-17 sites. The distance from sampling sites to the nearest ports varied in broad range (0.11 - 82 km) but on average the distance from port to the corresponding sites ranges in more narrow limits (17 - 26 km, Fig ++, A).  --> -->


<!-- ```{r Table_with_Ports_coordinates, eval=FALSE} -->

<!-- ports %>%  -->
<!--   select(-Shore) %>%  -->
<!--   kable(col.names = c("Port", "Status", "Latitude", "Longitude"), caption = "STable +. Ports presented in the Kandalaksha bay") -->
<!-- ``` -->

<!-- \pagebreak -->

<!-- ### Associations between environmental parameters -->

<!-- Pearson correlation between environmental parameters -->
<!-- ```{r Pearson_correlation_between_environmental_parameters} -->
<!-- # names(myt_site) -->
<!-- cor_df <-  -->
<!-- myt_site %>% select(Salinity, Min_dist_river, Min_dist_port,Fetch) %>% as.data.frame(.) %>%  cor(., method = "pearson") %>% as.data.frame(.) %>% round(., 3) %>% mutate(Predictor = names(.))   -->

<!-- cor_df <- cor_df[-4,] -->
<!-- cor_df[2:3, 1] <- NA -->
<!-- cor_df[3, 2] <- NA -->

<!-- cor_df <- cor_df %>% select(Predictor, Min_dist_river, Min_dist_port, Fetch) -->

<!-- ft_cor <- flextable(cor_df) -->

<!-- ft_cor -->

<!-- # cor.test(myt_site$Min_dist_river, myt_site$Min_dist_port) -->
<!-- ``` -->

<!-- \pagebreak -->

<!-- ```{r Rivers_parameters} -->

<!-- site_river <- -->
<!--   myt_site %>%  -->
<!--   group_by(River) %>%  -->
<!--   summarise(Site_Number = n()) -->



<!-- river_full <- -->
<!--   merge(river_full, site_river, all.x = T) %>%  -->
<!--   relocate(Site_Number, .after = Lon) %>%  -->
<!--   mutate(Site_Number = replace_na(Site_Number, replace = 0)) -->


<!-- ft_river <-  -->
<!-- river_full %>%  -->
<!--   # filter(River %in% unique(myt_full$River)) %>%  -->
<!--   select(-Shore, -Source) %>%  -->
<!--   flextable(.) -->

<!-- headers <- c("River", "Drainage Area sq. km", "River Size", "Mouth Latitude", "Mauth Longitude", "Number of sampling sites near the river") -->


<!-- ft_river <- -->
<!--   ft_river %>%  -->
<!--   set_header_labels(values = headers) %>%  -->
<!--   set_caption(caption = "Table +. Parameters of river and other fresh-water discharging source associated with Kandalaksha Bay.") -->

<!-- ft_river -->

<!-- ``` -->

<!-- \pagebreak -->





<!-- ```{r SFigure_preparation} -->
<!-- text_size <- 9 -->

<!-- theme_set(theme_bw() + theme(panel.grid = element_blank(), axis.title = element_text(size = text_size), axis.text.x = element_text(size = text_size))) -->

<!-- port_neghbourds <- myt_site %>% group_by(Port) %>% summarise(Site_Number = n(), Median_Dist = median(Min_dist_port)) -->

<!-- # range(round(port_neghbourds$Median_Dist)) -->

<!-- Pl_Ports <-  -->
<!--   myt_site %>%  -->
<!--   ggplot(., aes(x = Port, y = Min_dist_port)) +  -->
<!--   geom_boxplot(fill = "gray80") +  -->
<!--   geom_text(data = port_neghbourds, aes(y = Median_Dist - 3, label = Site_Number)) +  -->
<!--   labs(x = "Ports", y = "Distance to the nearest port (km)") + -->
<!--   theme(axis.text.x = element_text(angle = 60, vjust = 0.6)) -->



<!-- sites_fetch_df2 <-  -->
<!--   sites_fetch_df %>%  -->
<!--   melt(., id.vars = c("Site", "Site_Exposition"), value.name = "Fetch", variable.name = "Direction") -->

<!-- sites_fetch_df2$Direction <- factor(sites_fetch_df2$Direction, labels = c("Nort", "East", "South", "West", "Average Fetch")) -->

<!-- sites_fetch_df2 <- sites_fetch_df2 %>% filter(Direction == "Fetch") -->

<!-- Pl_Fetch <- -->
<!--   ggplot(sites_fetch_df2, aes(x = Site_Exposition, y = Fetch)) +  -->
<!--   geom_boxplot() +  -->
<!--   # scale_fill_manual(values = c("gray95", "gray80", "gray70", "gray60", "gray40")) +  -->
<!--   labs(x = "Visual assessment of \nsite exposition", y = "Wind fetch (km)") + -->
<!--   theme(axis.text.x = element_text(angle = 60, vjust = 0.6)) -->
<!-- # theme(legend.position = c(0.7, 0.75)) + -->



<!-- # range(myt_site$Salinity) -->

<!-- # median(myt_site$Salinity) -->


<!-- Pl_Salinity <- -->
<!--   myt_full %>% group_by(Site, River_Size) %>%  -->
<!--   summarise(Salinity = mean(Salinity)) %>%  -->
<!--   ggplot(., aes(x = River_Size, y = Salinity)) +  -->
<!--   geom_violin(fill = "gray") + -->
<!--   labs(x = "The size of the nearest river", y = "Salinity") + -->
<!--   theme(axis.text.x = element_text(angle = 60, vjust = 0.6)) -->


<!-- Pl_Salinity_vs_Dist <-  -->
<!--   ggplot(myt_site, aes(x = Min_dist_river, y = Salinity)) +  -->
<!--   geom_point() + -->
<!--   labs(x = "Distfnce to the nearest River", y = "Salinity") -->


<!-- Pl_Salinity_vs_Dist_to_Large_riv <-  -->
<!--   ggplot(myt_site, aes(x = Min_dist_river_Large, y = Salinity)) +  -->
<!--   geom_point() + -->
<!--   labs(x = "Distfnce to the nearest Large River", y = "Salinity") -->


<!-- Pl_Fetch_vs_Dist_to_Port <-  -->
<!--   ggplot(myt_site, aes(x = Min_dist_port, y = Fetch)) +  -->
<!--   geom_point() + -->
<!--   labs(x = "Distance to the nearest port", y = "Wind Fetch") -->


<!-- Pl_Salinity_vs_Dist_to_Port <-  -->
<!--   ggplot(myt_site, aes(x = Min_dist_port, y = Salinity)) +  -->
<!--   geom_point() + -->
<!--   labs(x = "Distance to the nearest port", y = "Salinity") -->


<!-- ``` -->




<!-- ```{r SFigure_Save, fig.width=8, fig.height=8, fig.cap="SFigure 1. Associations between environmental factors. "} -->

<!-- Pl_predictors <- -->
<!-- (Pl_Ports + Pl_Fetch) / -->
<!-- (Pl_Salinity + Pl_Salinity_vs_Dist) / -->
<!-- (Pl_Salinity_vs_Dist_to_Port + Pl_Fetch_vs_Dist_to_Port) + plot_annotation(tag_levels = "A") -->

<!-- ggsave(plot = Pl_predictors, filename = "figures/Pl_predictors.jpg", device = "jpeg", dpi = 600, width = 20, height = 20, units = "cm") -->


<!-- ``` -->


<!-- <!-- ![SFigure +. Associations between environmental factors. (A) Boxplots showing the variation in distances to the nearest port. Numbers insite the box indicate the number of sites for which the corresponding port is the nearest. (B) Boxplots showing the variation of wind fetch for two groups of sites (Exposed vs Sheltered) choosen for material collection. (C) The kernel density violin-plots showing salinity variation for sites near Small and Large rivers. (D) The association of salinity with distance to the mouth of the nearest river. (E) The association of salinity with distance to the nearest port. (F) The association between wind fetch and distance to the nearest port.](figures/Pl_predictors.jpg) --> -->


<!-- ```{r SFigure_plot,  fig.cap= "SFigure +. Associations between environmental factors. (A) Boxplots showing the variation in distances to the nearest port. Numbers insite the box indicate the number of sites for which the corresponding port is the nearest. (B) Boxplots showing the variation of wind fetch for two groups of sites (Exposed vs Sheltered) choosen for material collection. (C) The kernel density violin-plots showing salinity variation for sites near Small and Large rivers. (D) The association of salinity with distance to the mouth of the nearest river. (E) The association of salinity with distance to the nearest port. (F) The association between wind fetch and distance to the nearest port", fig.width=8, fig.height=8} -->

<!-- Pl_predictors -->
<!-- ``` -->




<!-- \pagebreak -->

<!-- ## Parameters of Model 2 -->

<!-- ```{r} -->

<!-- Model_2_summ_smoothers <- tidy(Mod_gam_reduced) -->

<!-- Model_2_summ_smoothers$term <- c( -->
<!--   "s(MinDistRiver)", -->
<!--   "s(AverageFetch)", -->
<!--   "s(MinDistPort)", -->
<!--   "Random effect s(Site)") -->

<!-- Model_2_summ_smoothers[, 2:4] <- round(Model_2_summ_smoothers[, 2:4], 1) -->
<!-- Model_2_summ_smoothers[, 5] <- round(Model_2_summ_smoothers[, 5], 4) -->


<!-- Model_2_summ_parametric <- tidy(Mod_gam_reduced, parametric = T) -->

<!-- Model_2_summ_parametric$term <- c("(Intercept)", -->
<!--                                   "Substrate(Algae)", -->
<!--                                   "RiverSize(Large)", -->
<!--                                   "PortStatus(Active)") -->

<!-- Model_2_summ_parametric[, 2:4] <- round(Model_2_summ_parametric[, 2:4], 1) -->
<!-- Model_2_summ_parametric[, 5] <- round(Model_2_summ_parametric[, 5], 4) -->

<!-- names(Model_2_summ_smoothers) <- names(Model_2_summ_parametric) -->


<!-- all_Model_2_summ <- rbind(Model_2_summ_smoothers, Model_2_summ_parametric) -->




<!-- ft_Model_2_summ_smoothers <- flextable(Model_2_summ_smoothers) -->

<!-- header <- c("Model term", "edf", "ref.edf", "Chi.sq", "p-value") -->

<!-- ft_Model_2_summ_smoothers <-  -->
<!--   ft_Model_2_summ_smoothers %>%  -->
<!--   set_header_labels(values = header) -->






<!-- ft_Model_2_summ_parametric <- flextable(Model_2_summ_parametric) -->
<!-- header2 <- c("Model term", "Parameter estimate", "SE", "z-statistic", "p-value") -->

<!-- ft_Model_2_summ_parametric <-  -->
<!--   ft_Model_2_summ_parametric %>%  -->
<!--   set_header_labels(values = header2) -->


<!-- ft_Model_2_summ_smoothers <- -->
<!--   ft_Model_2_summ_smoothers %>%  -->
<!--   set_caption(caption = "STable + . Parameters of smoothers and coefficients of parametric terms for the Model 2 describing dependency of PT on predictors (Salinity was removed).") -->

<!-- ft_Model_2_summ_smoothers -->
<!-- ft_Model_2_summ_parametric -->
<!-- ``` -->

<!-- \pagebreak -->

<!-- ## Parameters of Model describing association with substrate type -->

<!-- ```{r} -->
<!-- summ_Mod_Dif$term <- c( -->
<!--   "s(PT)", -->
<!--   "s(Total abundance)") -->

<!-- summ_Mod_Dif[, 2:4] <- round(summ_Mod_Dif[, 2:4], 1) -->
<!-- summ_Mod_Dif[, 5] <- round(summ_Mod_Dif[, 5], 4) -->


<!-- ft_summ_Mod_Dif <- flextable(summ_Mod_Dif) -->

<!-- header <- c("Model term", "edf", "ref.edf", "Chi.sq", "p-value") -->

<!-- ft_summ_Mod_Dif <-  -->
<!--   ft_summ_Mod_Dif %>%  -->
<!--   set_header_labels(values = header) -->

<!-- ft_summ_Mod_Dif <-  -->
<!--   ft_summ_Mod_Dif %>%  -->
<!--   set_caption(caption = "STable + . Parameters of smoothers in the model describing dependency of Difference of PT on Algae and Bottom substrates on proporion of T-morphotype and total mussel abundance in sites.") -->

<!-- ft_summ_Mod_Dif -->
<!-- ``` -->

